// @shipengine/connect-carrier-api/lib/models/service-points/service-point-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServicePointSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const service_point_feature_1 = require("../../app/metadata/service-point-feature");
const service_point_location_type_1 = require("../../app/metadata/service-point-location-type");
const service_point_response_address_schema_1 = require("./service-point-response-address-schema");
const weekly_hours_of_operation_schema_1 = require("./weekly-hours-of-operation-schema");
exports.ServicePointSchema = joi_1.default.object({
    service_point_id: joi_1.default.string().required(),
    address: service_point_response_address_schema_1.ServicePointResponseAddressSchema.required(),
    hours_of_operation: weekly_hours_of_operation_schema_1.WeeklyHoursOfOperationSchema.required(),
    features: joi_1.default.array().items(service_point_feature_1.ServicePointFeaturesEnumSchema).optional(),
    type: service_point_location_type_1.ServicePointLocationTypeEnumSchema.optional(),
});
