// @shipengine/connect-runtime/lib/errors/error-mapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mapBaseError = void 0;
const serialize_error_1 = require("serialize-error");
const mapDetailedError = (detail) => {
    return {
        external_error_code: detail.externalErrorCode,
        message: detail.message,
        external_http_status_code: detail.externalHttpStatusCode,
        raw_external_context: detail.externalContext
            ? JSON.stringify((0, serialize_error_1.serializeError)(detail.externalContext))
            : undefined,
        standardized_error_code: detail.errorCode,
    };
};
const getDetailedErrors = (error) => {
    let response = [];
    if (error.details) {
        response = error.details.map(mapDetailedError);
    }
    response.push({
        message: error.message,
    });
    return response;
};
const mapBaseError = (transactionId, error) => {
    return {
        transaction_id: transactionId,
        message: error.message,
        detailed_errors: getDetailedErrors(error),
        retry_after_seconds: error.retryAfterSeconds,
        retry_after_time: error.retryAfterTime,
        throttling_context: error.throttlingContext,
    };
};
exports.mapBaseError = mapBaseError;
