// @shipengine/connect-carrier-api/lib/models/inconterms/terms-of-trade-code.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TermsOfTradeCode = void 0;
/** @description Specifies the supported terms of trade codes (incoterms)*/
var TermsOfTradeCode;
(function (TermsOfTradeCode) {
    /** @description EX-WORKS incoterm  */
    TermsOfTradeCode["EXW"] = "exw";
    /** @description FREE CARRIER incoterm  */
    TermsOfTradeCode["FCA"] = "fca";
    /** @description CARRIAGE PAID TO incoterm */
    TermsOfTradeCode["CPT"] = "cpt";
    /** @description CARRIAGE AND INSURANCE PAID TO incoterm  */
    TermsOfTradeCode["CIP"] = "cip";
    /** @description DELIVERED PLACE UNLOADED incoterm, old name: DAT  */
    TermsOfTradeCode["DPU"] = "dpu";
    /** @description DELIVERED AT PLACE incoterm*/
    TermsOfTradeCode["DAP"] = "dap";
    /** @description DELIVERED DUTY PAID incoterm*/
    TermsOfTradeCode["DDP"] = "ddp";
    /** @description FREE ALONGSIDE SHIP incoterm (incoterm used for sea/inland waterway transport)     */
    TermsOfTradeCode["FAS"] = "fas";
    /** @description FREE ON BOARD incoterm (incoterm used for sea/inland waterway transport)     */
    TermsOfTradeCode["FOB"] = "fob";
    /** @description COST AND FREIGHT incoterm (incoterm used for sea/inland waterway transport)     */
    TermsOfTradeCode["CFR"] = "cfr";
    /** @description COST INSURANCE AND FREIGHT incoterm (incoterm used for sea/inland waterway transport)    */
    TermsOfTradeCode["CIF"] = "cif";
    /** @description DELIVERED DUTY UNPAID incoterm (incoterm retired in 2010)     */
    TermsOfTradeCode["DDU"] = "ddu";
    /** @description DELIVERED AT FRONTIER incoterm (incoterm retired in 2010)     */
    TermsOfTradeCode["DAF"] = "daf";
    /** @description DELIVERED EX QUAY incoterm (incoterm retired in 2010)     */
    TermsOfTradeCode["DEQ"] = "deq";
    /** @description DELIVERED EX SHIP incoterm (incoterm retired in 2010)     */
    TermsOfTradeCode["DES"] = "des";
})(TermsOfTradeCode || (exports.TermsOfTradeCode = TermsOfTradeCode = {}));
