// @shipengine/connect-carrier-api/lib/models/billing/billing-categories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingCategories = void 0;
/** @description Category for a billing line item */
var BillingCategories;
(function (BillingCategories) {
    BillingCategories["Uncategorized"] = "uncategorized";
    BillingCategories["Shipping"] = "shipping";
    BillingCategories["Insurance"] = "insurance";
    /** @description Charge for delivery confirmation, e.g. requiring adult signature */
    BillingCategories["Confirm"] = "confirm";
    BillingCategories["Discount"] = "discount";
    BillingCategories["FuelCharge"] = "fuel_charge";
    BillingCategories["AdditionalFees"] = "additional_fees";
    BillingCategories["Tariff"] = "tariff";
    BillingCategories["Tax"] = "tax";
    BillingCategories["Delivery"] = "delivery";
    BillingCategories["Handling"] = "handling";
    BillingCategories["SpecialGoods"] = "special_goods";
    BillingCategories["Pickup"] = "pickup";
    /** @description Charge for the location of the origin or destination */
    BillingCategories["LocationFee"] = "location_fee";
    /** @description Charge for non-standard sizes */
    BillingCategories["Oversize"] = "oversize";
    BillingCategories["Returns"] = "returns";
    BillingCategories["Notifications"] = "notifications";
    /** @description Amount to be paid to the driver, used for local delivery and courier services */
    BillingCategories["Tip"] = "tip";
})(BillingCategories || (exports.BillingCategories = BillingCategories = {}));
