// connect-carrier-api-utils/lib/tests-helper/enums/countries.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Countries = void 0;
/**
 * Countries used to select default address for specified country in AddressBuilder
 */
var Countries;
(function (Countries) {
    Countries["DEFAULT"] = "DEFAULT";
    Countries["PL"] = "PL";
    Countries["GB"] = "GB";
    Countries["AL"] = "AL";
    Countries["AR"] = "AR";
    Countries["AU"] = "AU";
    Countries["AM"] = "AM";
    Countries["AG"] = "AG";
    Countries["AT"] = "AT";
    Countries["BB"] = "BB";
    Countries["BY"] = "BY";
    Countries["BE"] = "BE";
    Countries["BZ"] = "BZ";
    Countries["BO"] = "BO";
    Countries["BA"] = "BA";
    Countries["BR"] = "BR";
    Countries["BG"] = "BG";
    Countries["CA"] = "CA";
    Countries["CL"] = "CL";
    Countries["CN"] = "CN";
    Countries["CO"] = "CO";
    Countries["HR"] = "HR";
    Countries["CU"] = "CU";
    Countries["CY"] = "CY";
    Countries["CZ"] = "CZ";
    Countries["DK"] = "DK";
    Countries["EC"] = "EC";
    Countries["EE"] = "EE";
    Countries["FO"] = "FO";
    Countries["FI"] = "FI";
    Countries["FR"] = "FR";
    Countries["GE"] = "GE";
    Countries["GF"] = "GF";
    Countries["DE"] = "DE";
    Countries["GI"] = "GI";
    Countries["GR"] = "GR";
    Countries["GT"] = "GT";
    Countries["GY"] = "GY";
    Countries["HT"] = "HT";
    Countries["HN"] = "HN";
    Countries["HU"] = "HU";
    Countries["IS"] = "IS";
    Countries["IN"] = "IN";
    Countries["IE"] = "IE";
    Countries["IT"] = "IT";
    Countries["JM"] = "JM";
    Countries["JP"] = "JP";
    Countries["KZ"] = "KZ";
    Countries["LV"] = "LV";
    Countries["LI"] = "LI";
    Countries["LU"] = "LU";
    Countries["MK"] = "MK";
    Countries["MT"] = "MT";
    Countries["MX"] = "MX";
    Countries["MD"] = "MD";
    Countries["MC"] = "MC";
    Countries["ME"] = "ME";
    Countries["NL"] = "NL";
    Countries["NZ"] = "NZ";
    Countries["NI"] = "NI";
    Countries["NG"] = "NG";
    Countries["NO"] = "NO";
    Countries["PA"] = "PA";
    Countries["PK"] = "PK";
    Countries["PY"] = "PY";
    Countries["PE"] = "PE";
    Countries["PT"] = "PT";
    Countries["PR"] = "PR";
    Countries["QA"] = "QA";
    Countries["KR"] = "KR";
    Countries["RO"] = "RO";
    Countries["RU"] = "RU";
    Countries["LC"] = "LC";
    Countries["SV"] = "SV";
    Countries["SM"] = "SM";
    Countries["RS"] = "RS";
    Countries["SK"] = "SK";
    Countries["SI"] = "SI";
    Countries["ES"] = "ES";
    Countries["SR"] = "SR";
    Countries["SE"] = "SE";
    Countries["CH"] = "CH";
    Countries["TR"] = "TR";
    Countries["UA"] = "UA";
    Countries["AE"] = "AE";
    Countries["US"] = "US";
    Countries["UY"] = "UY";
    Countries["VA"] = "VA";
    Countries["VE"] = "VE";
    Countries["AD"] = "AD";
    Countries["LT"] = "LT";
    Countries["IQ"] = "IQ";
    Countries["AF"] = "AF";
    Countries["IL"] = "IL";
    Countries["XA"] = "XA";
    Countries["SG"] = "SG";
})(Countries = exports.Countries || (exports.Countries = {}));
