// @shipengine/connect-carrier-api/lib/responses/create-manifest-response-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateManifestResponseSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const base_response_schema_1 = require("./base-response-schema");
const manifest_status_schema_1 = require("../models/manifest-status-schema");
const manifest_error_schema_1 = require("../models/manifests/manifest-error-schema");
const manifest_schema_1 = require("../models/manifests/manifest-schema");
exports.CreateManifestResponseSchema = base_response_schema_1.BaseResponseSchema.keys({
    transaction_id: joi_1.default.string().required(),
    manifests: joi_1.default.array().optional().items(manifest_schema_1.ManifestSchema),
    errors: manifest_error_schema_1.ManifestErrorSchema.optional(),
    manifest_request_id: joi_1.default.number().optional(),
    manifest_status: manifest_status_schema_1.ManifestStatusSchema.optional,
    total_shipments: joi_1.default.number().optional(),
    total_invalid: joi_1.default.number().optional(),
    manifest_identifiers: joi_1.default.any().optional(),
});
