// @shipengine/connect-carrier-api/lib/responses/create-label-response-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateLabelResponseSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const base_response_schema_1 = require("./base-response-schema");
const billing_line_item_schema_1 = require("../models/billing/billing-line-item-schema");
const document_schema_1 = require("../models/documents/document-schema");
const identifier_schema_1 = require("../models/identifier-schema");
const form_download_schema_1 = require("../models/labels/form-download-schema");
const label_download_schema_1 = require("../models/labels/label-download-schema");
const label_package_schema_1 = require("../models/labels/label-package-schema");
const relay_point_details_schema_1 = require("../models/relay-points/relay-point-details-schema");
const time_window_schema_1 = require("../models/time-window-schema");
const paperless_details_schema_1 = require("../models/labels/paperless-details-schema");
exports.CreateLabelResponseSchema = base_response_schema_1.BaseResponseSchema.keys({
    transaction_id: joi_1.default.string().required(),
    label_download: label_download_schema_1.LabelDownloadSchema.optional(),
    form_download: form_download_schema_1.FormDownloadSchema.optional(),
    paperless_details: paperless_details_schema_1.PaperlessDetailsSchema.optional(),
    documents: joi_1.default.array().optional().items(document_schema_1.DocumentSchema),
    packages: joi_1.default.array().optional().items(label_package_schema_1.LabelPackageSchema),
    billing_line_items: joi_1.default.array().optional().items(billing_line_item_schema_1.BillingLineItemSchema),
    tracking_number: joi_1.default.string().optional().empty(),
    tracking_url: joi_1.default.string().optional().empty(),
    trackable: joi_1.default.boolean().optional(),
    alternative_identifiers: joi_1.default.array().optional().items(identifier_schema_1.IdentifierSchema),
    estimated_delivery_datetime: joi_1.default.string().optional().empty().isoDate(),
    consolidator_service_code: joi_1.default.string().optional().empty(),
    carrier_pickup_window: time_window_schema_1.TimeWindowSchema.optional(),
    delivery_window: time_window_schema_1.TimeWindowSchema.optional(),
    relay_points: relay_point_details_schema_1.RelayPointDetailsSchema.optional(),
});
