// @shipengine/connect-carrier-api/lib/responses/schedule-pickup-response-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulePickupResponseSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const billing_line_item_schema_1 = require("../models/billing/billing-line-item-schema");
const pickup_confirmation_schema_1 = require("../models/pickup/pickup-confirmation-schema");
const pickup_window_schema_1 = require("../models/pickup/pickup-window-schema");
const base_response_schema_1 = require("./base-response-schema");
exports.SchedulePickupResponseSchema = base_response_schema_1.BaseResponseSchema.keys({
    confirmation: pickup_confirmation_schema_1.PickupConfirmationSchema.optional(),
    pickup_windows: joi_1.default.array().optional().items(pickup_window_schema_1.PickupWindowSchema),
    remarks: joi_1.default.string().optional().empty(),
    custom_properties: joi_1.default.object().pattern(/.*/, [joi_1.default.string()]).optional(),
    billing_line_items: joi_1.default.array().optional().items(billing_line_item_schema_1.BillingLineItemSchema),
});
