// @shipengine/connect-carrier-api/lib/models/rates/rate-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RateSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const billing_line_item_schema_1 = require("../billing/billing-line-item-schema");
const time_window_schema_1 = require("../time-window-schema");
exports.RateSchema = joi_1.default.object({
    service_code: joi_1.default.string().optional().empty(),
    ship_datetime: joi_1.default.string().optional().empty().isoDate(),
    estimated_delivery_datetime: joi_1.default.string().optional().empty().isoDate(),
    billing_line_items: joi_1.default.array().optional().items(billing_line_item_schema_1.BillingLineItemSchema),
    error_messages: joi_1.default.array().optional().items(joi_1.default.string()),
    warning_messages: joi_1.default.array().optional().items(joi_1.default.string()),
    negotiated_rate: joi_1.default.boolean().optional(),
    carrier_pickup_window: time_window_schema_1.TimeWindowSchema.optional(),
    delivery_window: time_window_schema_1.TimeWindowSchema.optional(),
    carrier_rate_id: joi_1.default.string().optional().empty(),
    expiration_datetime: joi_1.default.string().optional().empty().isoDate(),
    estimated_delivery_text: joi_1.default.string().optional().empty(),
    package_type: joi_1.default.string().optional().empty(),
    zone: joi_1.default.string().optional().empty(),
});
