// @shipengine/connect-carrier-api/lib/models/documents/document-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentType = void 0;
/** @description Types of documents */
var DocumentType;
(function (DocumentType) {
    /** @description A shipping label */
    DocumentType["Label"] = "label";
    /** @description A form that lists the details of goods that are being imported or exported when a citizen or visitor enters a customs territory (country's borders) */
    DocumentType["CustomsForm"] = "customs_form";
    /** @description A legal document issued by the seller (exporter) to the buyer (importer) in an international transaction and serves as a contract and a proof of sale between the buyer and seller */
    DocumentType["CommercialInvoice"] = "commercial_invoice";
    /** @description A legal document issued by a carrier to a shipper that details the type, quantity, and destination of the goods being carried (does not indicate the ownership of goods nor does it carry a title to the goods being sold) */
    DocumentType["BillOfLading"] = "bill_of_lading";
    /** @description A paperless label. For example: a paperless label that customers can display on their phone */
    DocumentType["PaperlessLabel"] = "paperless_label";
})(DocumentType || (exports.DocumentType = DocumentType = {}));
