// @shipengine/connect-carrier-api/lib/models/taxes/tax-identifier-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxIdentifierType = void 0;
/** @description Specifies the supported tax identification types  */
var TaxIdentifierType;
(function (TaxIdentifierType) {
    /** @description Tax Identification Number  */
    TaxIdentifierType["TIN"] = "tin";
    /** @description Employer Identification Number  */
    TaxIdentifierType["EIN"] = "ein";
    /** @description Social Security Number  */
    TaxIdentifierType["SSN"] = "ssn";
    /** @description Value Added Tax Identification Number  */
    TaxIdentifierType["VAT"] = "vat";
    /** @description Economic Operators Registration and Identification Number  */
    TaxIdentifierType["EORI"] = "eori";
    /** @description Import One-Stop Shop Number
     * The IOSS allows suppliers and electronic interfaces selling imported goods to buyers in the EU to
     * collect, declare and pay the VAT to the tax authorities, instead of making the buyer pay the VAT at the
     * moment the goods are imported into the EU as it was previously the case (for products over 22 EUR).
     * https://ec.europa.eu/taxation_customs/business/vat/ioss_en
     */
    TaxIdentifierType["IOSS"] = "ioss";
    /** @description Permanent Account Number
     * A ten-character alphanumeric identifier, issued by the Indian Income Tax
     */
    TaxIdentifierType["PAN"] = "pan";
    /** @description Norway's VAT on E-Commerce number
     * https://www.skatteetaten.no/en/business-and-organisation/vat-and-duties/vat/foreign/e-commerce-voec/
     */
    TaxIdentifierType["VOEC"] = "voec";
    /** @description Personal Customs Clearance Code (PCCC)*/
    TaxIdentifierType["PCCC"] = "pccc";
    /** @description One-Stop Shop (OSS).*/
    TaxIdentifierType["OSS"] = "oss";
    /** @description Passport number*/
    TaxIdentifierType["PASSPORT"] = "passport";
    /** @description Australian Business Number*/
    TaxIdentifierType["ABN"] = "abn";
})(TaxIdentifierType || (exports.TaxIdentifierType = TaxIdentifierType = {}));
