// @shipengine/connect-carrier-api/lib/app/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiEndpoints = void 0;
var ApiEndpoints;
(function (ApiEndpoints) {
    ApiEndpoints["Register"] = "/Register";
    ApiEndpoints["CreateLabel"] = "/CreateLabel";
    ApiEndpoints["VoidLabels"] = "/VoidLabels";
    ApiEndpoints["GetRates"] = "/GetRates";
    ApiEndpoints["CreateManifest"] = "/CreateManifest";
    ApiEndpoints["GetManifest"] = "/GetManifest";
    ApiEndpoints["SchedulePickup"] = "/SchedulePickup";
    ApiEndpoints["CancelPickup"] = "/CancelPickup";
    ApiEndpoints["Track"] = "/Track";
    ApiEndpoints["CreateNotification"] = "/CreateNotification";
    ApiEndpoints["CancelNotification"] = "/CancelNotification";
    ApiEndpoints["ValidateInboundData"] = "/ValidateInboundData";
    ApiEndpoints["NormalizeTrackingData"] = "/NormalizeTrackingData";
    ApiEndpoints["GetRelayPoints"] = "/GetRelayPoints";
    ApiEndpoints["GetServicePoints"] = "/GetServicePoints";
    ApiEndpoints["GetServicePoint"] = "/GetServicePoint";
    ApiEndpoints["ImportTrackingEvents"] = "/ImportTrackingEvents";
    ApiEndpoints["UpdateSettings"] = "/UpdateSettings";
})(ApiEndpoints || (exports.ApiEndpoints = ApiEndpoints = {}));
