// connect-carrier-api-utils/lib/validations/result/result.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Result = void 0;
var ResultStatus;
(function (ResultStatus) {
    ResultStatus[ResultStatus["Success"] = 0] = "Success";
    ResultStatus[ResultStatus["Error"] = 1] = "Error";
})(ResultStatus || (ResultStatus = {}));
class Result {
    constructor() {
        this.status = ResultStatus.Success;
        this.messages = [];
    }
    addResult(result) {
        if (result.status === ResultStatus.Error) {
            this.status = ResultStatus.Error;
            this.messages.push(result.getMessages());
        }
    }
    addError(message) {
        this.status = ResultStatus.Error;
        this.messages.push(message);
    }
    hasError() {
        return this.status === ResultStatus.Error || this.messages.length > 0;
    }
    getMessages() {
        return this.messages.join('\n');
    }
    static Success() {
        return new Result();
    }
    static Error(message) {
        const result = new Result();
        result.status = ResultStatus.Error;
        result.messages.push(message);
        return result;
    }
}
exports.Result = Result;
