// @shipengine/connect-carrier-api/lib/models/errors/standardized-error-code.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardizedErrorCode = void 0;
/** @deprecated Please Use Runtime Errors */
var StandardizedErrorCode;
(function (StandardizedErrorCode) {
    StandardizedErrorCode[StandardizedErrorCode["Generic"] = 0] = "Generic";
    StandardizedErrorCode[StandardizedErrorCode["Serialization"] = 1] = "Serialization";
    StandardizedErrorCode[StandardizedErrorCode["UnhandledModuleException"] = 2] = "UnhandledModuleException";
    StandardizedErrorCode[StandardizedErrorCode["Validation"] = 3] = "Validation";
    StandardizedErrorCode[StandardizedErrorCode["ExternalBadRequest"] = 400] = "ExternalBadRequest";
    StandardizedErrorCode[StandardizedErrorCode["ExternalUnauthorized"] = 401] = "ExternalUnauthorized";
    StandardizedErrorCode[StandardizedErrorCode["ExternalUnprocessableEntity"] = 422] = "ExternalUnprocessableEntity";
    StandardizedErrorCode[StandardizedErrorCode["ExternalTooManyRequests"] = 429] = "ExternalTooManyRequests";
    StandardizedErrorCode[StandardizedErrorCode["ExternalNotImplemented"] = 501] = "ExternalNotImplemented";
    StandardizedErrorCode[StandardizedErrorCode["ExternalBadGateway"] = 502] = "ExternalBadGateway";
    StandardizedErrorCode[StandardizedErrorCode["ExternalServiceUnavailable"] = 503] = "ExternalServiceUnavailable";
    StandardizedErrorCode[StandardizedErrorCode["ExternalGatewayTimeout"] = 504] = "ExternalGatewayTimeout";
    StandardizedErrorCode[StandardizedErrorCode["ExternalServerError"] = 520] = "ExternalServerError";
})(StandardizedErrorCode || (exports.StandardizedErrorCode = StandardizedErrorCode = {}));
