// connect-carrier-api-utils/lib/tests-helper/fake-address-builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FakeAddressBuilder = void 0;
const connect_carrier_api_1 = require("@shipengine/connect-carrier-api");
const tests_data_source_1 = require("./tests-data-source");
const helper_1 = require("../helpers/helper");
class FakeAddressBuilder {
    constructor() {
        this.Build = () => {
            return this.address;
        };
        this.SetCountry = (value) => {
            this.address.country_code = value;
            return this;
        };
        this.SetAddressLines = (value) => {
            this.address.address_lines = value;
            return this;
        };
        this.SetPostalCode = (value) => {
            this.address.postal_code = value;
            return this;
        };
        this.SetAddressResidentialIndicator = (value) => {
            this.address.address_residential_indicator = value;
            return this;
        };
        this.SetCity = (value) => {
            this.address.city_locality = value;
            return this;
        };
        this.SetCompanyName = (value) => {
            this.address.company_name = value;
            return this;
        };
        this.SetEmail = (value) => {
            this.address.email = value;
            return this;
        };
        this.SetFirstName = (value) => {
            this.address.first_name = value;
            return this;
        };
        this.SetLastName = (value) => {
            this.address.last_name = value;
            return this;
        };
        this.SetInstructions = (value) => {
            this.address.instructions = value;
            return this;
        };
        this.SetIsEu = (value) => {
            this.address.is_eu = value;
            return this;
        };
        this.SetName = (value) => {
            this.address.name = value;
            return this;
        };
        this.SetPhoneNumber = (value) => {
            this.address.phone_number = value;
            return this;
        };
        this.SetStateProvince = (value) => {
            this.address.state_province = value;
            return this;
        };
        this.SetDefaultAddressForSpecifiedCountry = (country) => {
            const address = (0, helper_1.GetSafeValueFromMap)(tests_data_source_1.TESTS_ADDRESS_DATA_SOURCE, country);
            this.address.country_code = address.country_code;
            this.address.address_lines = address.address_lines;
            this.address.postal_code = address.postal_code;
            this.address.address_residential_indicator = address.address_residential_indicator;
            this.address.city_locality = address.city_locality;
            this.address.company_name = address.company_name;
            this.address.email = address.email;
            this.address.first_name = address.first_name;
            this.address.last_name = address.last_name;
            this.address.instructions = address.instructions;
            this.address.is_eu = address.is_eu;
            this.address.name = address.name;
            this.address.phone_number = address.phone_number;
            this.address.state_province = address.state_province;
            return this;
        };
        this.address = {
            country_code: '',
            address_lines: [],
            postal_code: '',
            address_residential_indicator: connect_carrier_api_1.AddressResidentialIndicator.No,
            city_locality: '',
            company_name: '',
            email: '',
            first_name: '',
            last_name: '',
            instructions: '',
            is_eu: false,
            name: '',
            phone_number: '',
            state_province: ''
        };
    }
}
exports.FakeAddressBuilder = FakeAddressBuilder;
