// connect-carrier-api-utils/lib/tests-helper/fake-custom-item-builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FakeCustomItemBuilder = void 0;
const connect_carrier_api_1 = require("@shipengine/connect-carrier-api");
class FakeCustomItemBuilder {
    constructor() {
        this.Build = () => {
            return this.customItem;
        };
        this.SetCountryOfOrigin = (value) => {
            this.customItem.country_of_origin = value;
            return this;
        };
        this.SetQuantity = (value) => {
            this.customItem.quantity = value;
            return this;
        };
        this.SetDescription = (value) => {
            this.customItem.description = value;
            return this;
        };
        this.SetSkuDescription = (value) => {
            this.customItem.sku_description = value;
            return this;
        };
        this.SetSku = (value) => {
            this.customItem.sku = value;
            return this;
        };
        this.SetDefault = () => {
            this.SetSku('123456');
            this.SetSkuDescription('Comics');
            this.SetDescription('Books');
            this.SetQuantity(1);
            this.SetCountryOfOrigin('Poland');
            return this;
        };
        this.customItem = {
            sku: '',
            country_of_origin: '',
            description: '',
            sku_description: '',
            quantity: 1,
            item_weight: {
                weight_in_grams: 100,
                source_weight_unit: connect_carrier_api_1.WeightUnit.Grams,
                source_weight: 100,
                weight_in_ounces: 20
            },
            value: {
                currency: 'USD',
                amount: '100'
            },
            harmonized_tariff_code: '123456'
        };
    }
}
exports.FakeCustomItemBuilder = FakeCustomItemBuilder;
