// connect-carrier-api-utils/lib/tests-helper/fake-create-label-request-builder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FakeCreateLabelRequestBuilder = void 0;
const fake_address_builder_1 = require("./fake-address-builder");
class FakeCreateLabelRequestBuilder {
    constructor() {
        this.Build = () => {
            return this.request;
        };
        this.SetSender = (sender) => {
            this.request.ship_from = sender;
            return this;
        };
        this.SetShipDateTime = (value) => {
            this.request.ship_datetime = value;
            return this;
        };
        this.SetReceiver = (receiver) => {
            this.request.ship_to = receiver;
            return this;
        };
        this.SetService = (service) => {
            this.request.service_code = service;
            return this;
        };
        this.SetPackages = (packages) => {
            this.request.packages = packages;
            return this;
        };
        this.SetTransactionId = (value) => {
            this.request.transaction_id = value;
            return this;
        };
        this.SetMetadata = (value) => {
            this.request.metadata = value;
            return this;
        };
        this.request = {
            is_return_label: false,
            packages: [],
            ship_datetime: '',
            ship_from: new fake_address_builder_1.FakeAddressBuilder().Build(),
            ship_to: new fake_address_builder_1.FakeAddressBuilder().Build(),
            transaction_id: ''
        };
    }
}
exports.FakeCreateLabelRequestBuilder = FakeCreateLabelRequestBuilder;
