// connect-carrier-api-utils/lib/tests-helper/tests-data-source.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LU_ADDRESS = exports.LI_ADDRESS = exports.LV_ADDRESS = exports.KZ_ADDRESS = exports.JP_ADDRESS = exports.JM_ADDRESS = exports.IT_ADDRESS = exports.IE_ADDRESS = exports.IN_ADDRESS = exports.IS_ADDRESS = exports.HU_ADDRESS = exports.HN_ADDRESS = exports.HT_ADDRESS = exports.GY_ADDRESS = exports.GT_ADDRESS = exports.GR_ADDRESS = exports.GI_ADDRESS = exports.DE_ADDRESS = exports.GF_ADDRESS = exports.GE_ADDRESS = exports.FR_ADDRESS = exports.FI_ADDRESS = exports.FO_ADDRESS = exports.EE_ADDRESS = exports.EC_ADDRESS = exports.DK_ADDRESS = exports.CZ_ADDRESS = exports.CY_ADDRESS = exports.CU_ADDRESS = exports.HR_ADDRESS = exports.CO_ADDRESS = exports.CN_ADDRESS = exports.CL_ADDRESS = exports.CA_ADDRESS = exports.BG_ADDRESS = exports.BR_ADDRESS = exports.BA_ADDRESS = exports.BO_ADDRESS = exports.BZ_ADDRESS = exports.BE_ADDRESS = exports.BY_ADDRESS = exports.BB_ADDRESS = exports.AT_ADDRESS = exports.AG_ADDRESS = exports.AM_ADDRESS = exports.AU_ADDRESS = exports.AR_ADDRESS = exports.AL_ADDRESS = exports.PL_ADDRESS = exports.DEFAULT_ADDRESS = void 0;
exports.TESTS_ADDRESS_DATA_SOURCE = exports.GB_ADDRESS = exports.AD_ADDRESS = exports.VE_ADDRESS = exports.VA_ADDRESS = exports.UY_ADDRESS = exports.US_ADDRESS = exports.AE_ADDRESS = exports.UA_ADDRESS = exports.TR_ADDRESS = exports.CH_ADDRESS = exports.SE_ADDRESS = exports.SR_ADDRESS = exports.ES_ADDRESS = exports.SI_ADDRESS = exports.SK_ADDRESS = exports.RS_ADDRESS = exports.SM_ADDRESS = exports.SV_ADDRESS = exports.LC_ADDRESS = exports.RU_ADDRESS = exports.RO_ADDRESS = exports.KR_ADDRESS = exports.QA_ADDRESS = exports.PR_ADDRESS = exports.PT_ADDRESS = exports.PE_ADDRESS = exports.PY_ADDRESS = exports.PK_ADDRESS = exports.PA_ADDRESS = exports.NO_ADDRESS = exports.NG_ADDRESS = exports.NI_ADDRESS = exports.NZ_ADDRESS = exports.NL_ADDRESS = exports.ME_ADDRESS = exports.MC_ADDRESS = exports.MD_ADDRESS = exports.MX_ADDRESS = exports.MT_ADDRESS = exports.MK_ADDRESS = void 0;
const connect_carrier_api_1 = require("@shipengine/connect-carrier-api");
const countries_1 = require("./enums/countries");
exports.DEFAULT_ADDRESS = {
    country_code: 'PL',
    address_lines: ['AddressLine1', 'AddressLine2', 'AddressLine3'],
    postal_code: '66-600',
    address_residential_indicator: connect_carrier_api_1.AddressResidentialIndicator.No,
    city_locality: 'Krosno Odrzanskie',
    company_name: 'XLogics4Ever',
    email: 'xlogics@test.com',
    first_name: 'Bombel',
    last_name: 'ek',
    instructions: 'Test',
    is_eu: true,
    name: 'Bombelek',
    phone_number: '111111111',
    state_province: ''
};
exports.PL_ADDRESS = {
    country_code: 'PL',
    address_lines: ['AddressLine1', 'AddressLine2', 'AddressLine3'],
    postal_code: '65-001',
    address_residential_indicator: connect_carrier_api_1.AddressResidentialIndicator.No,
    city_locality: 'Zielona Góra',
    company_name: 'Poland Company',
    email: 'polandcompany@test.com',
    first_name: 'Jan',
    last_name: 'Kowalski',
    instructions: 'Instructions',
    is_eu: true,
    name: 'Jan Kowalski',
    phone_number: '68123456789',
    state_province: ''
};
exports.AL_ADDRESS = {
    first_name: 'Kristi',
    country_code: 'AL',
    address_lines: [
        'Rruga Sami Frashëri',
        '36',
        '\n'
    ],
    postal_code: '1000',
    address_residential_indicator: undefined,
    city_locality: 'Tirana',
    company_name: 'Company Kristi',
    email: 'vataK@test.com',
    last_name: 'Vata',
    instructions: '',
    is_eu: false,
    name: 'Kristi Vata',
    phone_number: '+355672054461',
    state_province: 'Tirana'
};
exports.AR_ADDRESS = {
    first_name: 'Willy',
    country_code: 'AR',
    address_lines: [
        'Rivadavia',
        '891',
        '\n'
    ],
    postal_code: '8363',
    address_residential_indicator: undefined,
    city_locality: 'Lamarque',
    company_name: 'Company Willy',
    email: 'willy.otamendi@test.com',
    last_name: 'Otamendi',
    instructions: '',
    is_eu: false,
    name: 'Willy Otamendi',
    phone_number: '+54 2946 49-7248',
    state_province: 'Río Negro'
};
exports.AU_ADDRESS = {
    first_name: 'Oliver',
    country_code: 'AU',
    address_lines: [
        'Elizabeth St',
        '68/1',
        '\n'
    ],
    postal_code: '2548',
    address_residential_indicator: undefined,
    city_locality: 'Merimbula',
    company_name: 'Company Oliver',
    email: '\n',
    last_name: 'Smith',
    instructions: '',
    is_eu: false,
    name: 'Oliver Smith',
    phone_number: '+61 2 6945 4394',
    state_province: 'NSW'
};
exports.AM_ADDRESS = {
    first_name: 'Ankine',
    country_code: 'AM',
    address_lines: [
        'Muratsan St',
        '115',
        '\n'
    ],
    postal_code: '0075',
    address_residential_indicator: undefined,
    city_locality: 'Yerevan',
    company_name: 'Company Ankine',
    email: 'ankineg19980@test.com',
    last_name: 'Ghazaryan ',
    instructions: '',
    is_eu: false,
    name: 'Ankine Ghazaryan ',
    phone_number: '+37410550230',
    state_province: 'Yerevan'
};
exports.AG_ADDRESS = {
    first_name: 'Joel',
    country_code: 'AG',
    address_lines: [
        'Perry Bay',
        '12',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: "St John's",
    company_name: 'Company Joel',
    email: 'joel.peters@test.com',
    last_name: 'Peters',
    instructions: '',
    is_eu: false,
    name: 'Joel Peters',
    phone_number: '+1 268-481-0000',
    state_province: 'Saint John'
};
exports.AT_ADDRESS = {
    first_name: 'Robert',
    country_code: 'AT',
    address_lines: [
        'Hoher Markt',
        '10-11',
        '\n'
    ],
    postal_code: '1010',
    address_residential_indicator: undefined,
    city_locality: 'Vienna',
    company_name: 'Company Robert',
    email: 'robert.hinteregger@test.com',
    last_name: 'Hinteregger',
    instructions: '',
    is_eu: false,
    name: 'Robert Hinteregger',
    phone_number: '+43 1 535 56 06',
    state_province: 'Vienna'
};
exports.BB_ADDRESS = {
    first_name: 'Liam',
    country_code: 'BB',
    address_lines: [
        'Highway',
        '3',
        '\n'
    ],
    postal_code: 'BB18075',
    address_residential_indicator: undefined,
    city_locality: 'Belair',
    company_name: 'Company Liam',
    email: 'liam.boxill@test.com',
    last_name: 'Boxill',
    instructions: '',
    is_eu: false,
    name: 'Liam Boxill',
    phone_number: '+1 246-822-0000',
    state_province: 'Saint George'
};
exports.BY_ADDRESS = {
    first_name: 'Aleksandr',
    country_code: 'BY',
    address_lines: [
        'Savieckaja',
        '3',
        '\n'
    ],
    postal_code: '231592',
    address_residential_indicator: undefined,
    city_locality: 'Masty',
    company_name: 'Company Aleksandr',
    email: 'aleksandr.gorbunov@test.com',
    last_name: 'Gorbunov',
    instructions: '',
    is_eu: false,
    name: 'Aleksandr Gorbunov',
    phone_number: '+375 214 12-20-60',
    state_province: 'Hrodzienskaja wobłasć'
};
exports.BE_ADDRESS = {
    first_name: 'Thibaut',
    country_code: 'BE',
    address_lines: [
        'Handelsstraat',
        '40',
        '\n'
    ],
    postal_code: '2060',
    address_residential_indicator: undefined,
    city_locality: 'Antwerpen',
    company_name: 'Company Thibaut',
    email: 'th.mignolet@test.com',
    last_name: 'Mignolet',
    instructions: '',
    is_eu: false,
    name: 'Thibaut Mignolet',
    phone_number: '+32 3 636 08 00',
    state_province: 'Antwerp'
};
exports.BZ_ADDRESS = {
    first_name: 'Woodrow',
    country_code: 'BZ',
    address_lines: [
        'Western Hwy',
        '61 Mile',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Georgeville',
    company_name: 'Company Woodrow',
    email: 'woodrow.westby@test.com',
    last_name: 'Westby',
    instructions: '',
    is_eu: false,
    name: 'Woodrow Westby',
    phone_number: '+501 824-0000',
    state_province: 'Cayo'
};
exports.BO_ADDRESS = {
    first_name: 'Carlos',
    country_code: 'BO',
    address_lines: [
        'Olañeta',
        '37',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Tarabuco',
    company_name: 'Company Carlos',
    email: 'carlos.cordano@test.com',
    last_name: 'Cordano',
    instructions: '',
    is_eu: false,
    name: 'Carlos Cordano',
    phone_number: '+591 65430000',
    state_province: 'Chuquisaca'
};
exports.BA_ADDRESS = {
    first_name: 'Vedad',
    country_code: 'BA',
    address_lines: [
        'Safvet-bega Bašagića',
        '55',
        '\n'
    ],
    postal_code: '71000',
    address_residential_indicator: undefined,
    city_locality: 'Sarajevo',
    company_name: 'Company Vedad',
    email: 'najicvedad@test.com',
    last_name: 'Bajić',
    instructions: '',
    is_eu: false,
    name: 'Vedad Bajić',
    phone_number: '+38733576675',
    state_province: 'Kanton Sarajevo'
};
exports.BR_ADDRESS = {
    first_name: 'Filipe',
    country_code: 'BR',
    address_lines: [
        'Travessa Cônego Domingos Maltez',
        '63',
        '\n'
    ],
    postal_code: '68950-000',
    address_residential_indicator: undefined,
    city_locality: 'Amapá',
    company_name: 'Company Filipe',
    email: 'filipe.danilo@test.com',
    last_name: 'Danilo',
    instructions: '',
    is_eu: false,
    name: 'Filipe Danilo',
    phone_number: '+55 96 3421-0000',
    state_province: 'Amapá'
};
exports.BG_ADDRESS = {
    first_name: 'Ivan',
    country_code: 'BG',
    address_lines: [
        'Aleksandar Stamboliyski',
        '101',
        '\n'
    ],
    postal_code: '1303',
    address_residential_indicator: undefined,
    city_locality: 'Sofia',
    company_name: 'Company Ivan',
    email: 'ivan.iliev@test.com',
    last_name: 'Iliev',
    instructions: '',
    is_eu: false,
    name: 'Ivan Iliev',
    phone_number: '+35929258519',
    state_province: '\n'
};
exports.CA_ADDRESS = {
    first_name: 'Herman',
    country_code: 'CA',
    address_lines: [
        'Bass Lake Rd',
        '500',
        '\n'
    ],
    postal_code: 'P0T 2M0',
    address_residential_indicator: undefined,
    city_locality: 'Shuniah',
    company_name: 'Company Herman',
    email: 'herman.murray@test.com',
    last_name: 'Murray',
    instructions: '',
    is_eu: false,
    name: 'Herman Murray',
    phone_number: '+1 807-622-6208',
    state_province: 'Ontario (ON)'
};
exports.CL_ADDRESS = {
    first_name: 'Vicente',
    country_code: 'CL',
    address_lines: [
        'San Andrés',
        '37',
        '\n'
    ],
    postal_code: '4030000',
    address_residential_indicator: undefined,
    city_locality: 'Concepción',
    company_name: 'Company Vicente',
    email: '\n',
    last_name: 'Hernández',
    instructions: '',
    is_eu: false,
    name: 'Vicente Hernández',
    phone_number: '+56 55 263 7589',
    state_province: 'Región del Bío Bio'
};
exports.CN_ADDRESS = {
    first_name: 'Lai Chun',
    country_code: 'CN',
    address_lines: [
        'North St',
        '16',
        '\n'
    ],
    postal_code: '100070',
    address_residential_indicator: undefined,
    city_locality: 'Beijing Shi',
    company_name: 'Company Lai Chun',
    email: 'LaiChunHo@test.com',
    last_name: 'Ho',
    instructions: '',
    is_eu: false,
    name: 'Lai Chun Ho',
    phone_number: '+86 16532924932',
    state_province: 'Beijing'
};
exports.CO_ADDRESS = {
    first_name: 'Santiago',
    country_code: 'CO',
    address_lines: [
        'Cra. 70',
        '#44B-66',
        '\n'
    ],
    postal_code: '050034',
    address_residential_indicator: undefined,
    city_locality: 'Medellín',
    company_name: 'Company Santiago',
    email: 'santiago212@test.com',
    last_name: 'Falcao',
    instructions: '',
    is_eu: false,
    name: 'Santiago Falcao',
    phone_number: '+5742600226',
    state_province: 'Antioquia'
};
exports.HR_ADDRESS = {
    first_name: 'Ivan',
    country_code: 'HR',
    address_lines: [
        'Bjelovarska',
        '48B',
        '\n'
    ],
    postal_code: '10010',
    address_residential_indicator: undefined,
    city_locality: 'Zagreb',
    company_name: 'Company Ivan',
    email: 'ivan.k@test.com',
    last_name: 'Kalinic',
    instructions: '',
    is_eu: false,
    name: 'Ivan Kalinic',
    phone_number: '+385 800 400 000',
    state_province: 'Zagreb'
};
exports.CU_ADDRESS = {
    first_name: 'Maykel',
    country_code: 'CU',
    address_lines: [
        'Gerardo Medina',
        '15',
        '\n'
    ],
    postal_code: '20100',
    address_residential_indicator: undefined,
    city_locality: 'Pinar del Río',
    company_name: 'Company Maykel',
    email: 'maykel.corrales@test.com',
    last_name: 'Corrales',
    instructions: '',
    is_eu: false,
    name: 'Maykel Corrales',
    phone_number: '+53 48 724947',
    state_province: 'Pinar del Río'
};
exports.CY_ADDRESS = {
    first_name: 'Janis',
    country_code: 'CY',
    address_lines: [
        'Venetiotissas Street',
        '1',
        '\n'
    ],
    postal_code: '8060',
    address_residential_indicator: undefined,
    city_locality: 'Poli Crysochous',
    company_name: 'Company Janis',
    email: 'janis.p@test.com',
    last_name: 'Pittas',
    instructions: '',
    is_eu: false,
    name: 'Janis Pittas',
    phone_number: '+357 26 323500',
    state_province: '\n'
};
exports.CZ_ADDRESS = {
    first_name: 'Pavel',
    country_code: 'CZ',
    address_lines: [
        'Jakubská',
        '2',
        '\n'
    ],
    postal_code: '110 00',
    address_residential_indicator: undefined,
    city_locality: 'Prague',
    company_name: 'Company Pavel',
    email: 'pavelnedved@test.com',
    last_name: 'Nedved',
    instructions: '',
    is_eu: false,
    name: 'Pavel Nedved',
    phone_number: '+420 222 316 328',
    state_province: 'Prague'
};
exports.DK_ADDRESS = {
    first_name: 'Simon',
    country_code: 'DK',
    address_lines: [
        'Jacob Gades Allé',
        '1',
        '\n'
    ],
    postal_code: '6600',
    address_residential_indicator: undefined,
    city_locality: 'Vejen',
    company_name: 'Company Simon',
    email: '\n',
    last_name: 'Jorgensen',
    instructions: '',
    is_eu: false,
    name: 'Simon Jorgensen',
    phone_number: '+45 75 36 17 00',
    state_province: 'Sydjylland'
};
exports.EC_ADDRESS = {
    first_name: 'Pedro',
    country_code: 'EC',
    address_lines: [
        'Isidro Ayora Cueva',
        '1',
        '\n'
    ],
    postal_code: '020555',
    address_residential_indicator: undefined,
    city_locality: 'Guayaquil',
    company_name: 'Company Pedro',
    email: 'pedro.aimar@test.com',
    last_name: 'Aimar',
    instructions: '',
    is_eu: false,
    name: 'Pedro Aimar',
    phone_number: '+593 96 991 0000',
    state_province: 'Bolívar'
};
exports.EE_ADDRESS = {
    first_name: 'Ragnar',
    country_code: 'EE',
    address_lines: [
        'Ehitajate',
        '27',
        '\n'
    ],
    postal_code: '12618',
    address_residential_indicator: undefined,
    city_locality: 'Tallinn',
    company_name: 'Company Ragnar',
    email: 'ragnar.klavan@test.com',
    last_name: 'Klavan',
    instructions: '',
    is_eu: false,
    name: 'Ragnar Klavan',
    phone_number: '+372 1314',
    state_province: 'Harju maakond'
};
exports.FO_ADDRESS = {
    first_name: 'Gunnar',
    country_code: 'FO',
    address_lines: [
        'Støðarvegur',
        '3',
        '\n'
    ],
    postal_code: '860',
    address_residential_indicator: undefined,
    city_locality: 'Sandvík',
    company_name: 'Company Gunnar',
    email: 'gunnar.hansen@test.com',
    last_name: 'Hansen',
    instructions: '',
    is_eu: false,
    name: 'Gunnar Hansen',
    phone_number: '+298 611010',
    state_province: '\n'
};
exports.FI_ADDRESS = {
    first_name: 'Janne',
    country_code: 'FI',
    address_lines: [
        'Korkeavuorenkatu',
        '23',
        '\n'
    ],
    postal_code: '00130',
    address_residential_indicator: undefined,
    city_locality: 'Helsinki',
    company_name: 'Company Janne',
    email: 'janne.asikainen@test.com',
    last_name: 'Asikainen',
    instructions: '',
    is_eu: false,
    name: 'Janne Asikainen',
    phone_number: '+358 9 6220540',
    state_province: 'Uusimaa'
};
exports.FR_ADDRESS = {
    first_name: "N'Golo",
    country_code: 'FR',
    address_lines: [
        'Boulevard Saint-Marcel',
        '73',
        '\n'
    ],
    postal_code: '75013',
    address_residential_indicator: undefined,
    city_locality: 'Paris',
    company_name: "Company N'Golo",
    email: 'ngolo.coman@test.com',
    last_name: 'Coman',
    instructions: '',
    is_eu: false,
    name: "N'Golo Coman",
    phone_number: '+33 1 40 79 92 79',
    state_province: 'Île-de-France'
};
exports.GE_ADDRESS = {
    first_name: 'ეთერი',
    country_code: 'GE',
    address_lines: [
        'რუსთაველის გამზირი',
        '26-28 ',
        '\n'
    ],
    postal_code: '6000',
    address_residential_indicator: undefined,
    city_locality: 'ბათუმი',
    company_name: 'Company ეთერი',
    email: '\n',
    last_name: 'აბაკელია',
    instructions: '',
    is_eu: false,
    name: 'ეთერი აბაკელია',
    phone_number: '+995 422 22 00 90',
    state_province: 'აჭარა'
};
exports.GF_ADDRESS = {
    first_name: 'Inrick',
    country_code: 'GF',
    address_lines: [
        'Rue de la Vieille Église',
        '6',
        '\n'
    ],
    postal_code: '97318',
    address_residential_indicator: undefined,
    city_locality: 'Mana',
    company_name: 'Company Inrick',
    email: 'inrick.adam@test.com',
    last_name: 'Adam',
    instructions: '',
    is_eu: false,
    name: 'Inrick Adam',
    phone_number: '+594 594 34 00 00',
    state_province: 'Mana'
};
exports.DE_ADDRESS = {
    first_name: 'Jürgen',
    country_code: 'DE',
    address_lines: [
        'Neustädtische Kirchstraße',
        '3',
        '\n'
    ],
    postal_code: '10117',
    address_residential_indicator: undefined,
    city_locality: 'Berlin',
    company_name: 'Company Jürgen',
    email: 'jurgen_schroder@test.com',
    last_name: 'Schröder',
    instructions: '',
    is_eu: false,
    name: 'Jürgen Schröder',
    phone_number: '+49 1570 751 131 ',
    state_province: 'Berlin'
};
exports.GI_ADDRESS = {
    first_name: 'Erin',
    country_code: 'GI',
    address_lines: [
        'Winston Churchill Ave',
        '20',
        '\n'
    ],
    postal_code: 'GX11 1AA',
    address_residential_indicator: undefined,
    city_locality: 'Gibraltar',
    company_name: 'Company Erin',
    email: 'erin.wiseman@test.com',
    last_name: 'Wiseman',
    instructions: '',
    is_eu: false,
    name: 'Erin Wiseman',
    phone_number: '+350 200 00000',
    state_province: '\n'
};
exports.GR_ADDRESS = {
    first_name: 'Georgios',
    country_code: 'GR',
    address_lines: [
        'Parasiou',
        '1',
        '\n'
    ],
    postal_code: '691 00',
    address_residential_indicator: undefined,
    city_locality: 'Komotini',
    company_name: 'Company Georgios',
    email: 'georgios.makos@test.com',
    last_name: 'Makos',
    instructions: '',
    is_eu: false,
    name: 'Georgios Makos',
    phone_number: '+30 2531 029680',
    state_province: 'Thráki'
};
exports.GT_ADDRESS = {
    first_name: 'Paulo',
    country_code: 'GT',
    address_lines: [
        '9 Calle',
        '5',
        '37'
    ],
    postal_code: '01007',
    address_residential_indicator: undefined,
    city_locality: 'Ciudad de Guatemala',
    company_name: 'Company Paulo',
    email: 'paulo.hagen@test.com',
    last_name: 'Hagen',
    instructions: '',
    is_eu: false,
    name: 'Paulo Hagen',
    phone_number: '+502 2440 0000',
    state_province: 'Guatemala'
};
exports.GY_ADDRESS = {
    first_name: 'Chris',
    country_code: 'GY',
    address_lines: [
        'Eping Ave',
        '8',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Georgetown',
    company_name: 'Company Chris',
    email: 'chris.durant@test.com',
    last_name: 'Durant',
    instructions: '',
    is_eu: false,
    name: 'Chris Durant',
    phone_number: '+592 227 0000',
    state_province: 'Mahaica-Berbice'
};
exports.HT_ADDRESS = {
    first_name: 'Johnny',
    country_code: 'HT',
    address_lines: [
        'Jean-Jacques Dessalines',
        '12',
        '\n'
    ],
    postal_code: 'HT 4310',
    address_residential_indicator: undefined,
    city_locality: 'Saint-Marc',
    company_name: 'Company Johnny',
    email: 'johnny.geffrard@test.com',
    last_name: 'Geffrard',
    instructions: '',
    is_eu: false,
    name: 'Johnny Geffrard',
    phone_number: '+509 34 41 0000',
    state_province: 'Artibonite'
};
exports.HN_ADDRESS = {
    first_name: 'Donis',
    country_code: 'HN',
    address_lines: [
        'GL26',
        '104',
        '\n'
    ],
    postal_code: '31301',
    address_residential_indicator: undefined,
    city_locality: 'Puerto Lempira',
    company_name: 'Company Donis',
    email: 'donis.lopez@test.com',
    last_name: 'Lopez',
    instructions: '',
    is_eu: false,
    name: 'Donis Lopez',
    phone_number: '+504 9723-0000',
    state_province: 'Gracias a Dios'
};
exports.HU_ADDRESS = {
    first_name: 'Adam',
    country_code: 'HU',
    address_lines: [
        'Ócsai út',
        '4',
        '\n'
    ],
    postal_code: '1239',
    address_residential_indicator: undefined,
    city_locality: 'Budapest',
    company_name: 'Company Adam',
    email: 'adam.kiraly@test.com',
    last_name: 'Kiraly',
    instructions: '',
    is_eu: false,
    name: 'Adam Kiraly',
    phone_number: '+36 1 289 0999',
    state_province: 'Budapest'
};
exports.IS_ADDRESS = {
    first_name: 'Indridi ',
    country_code: 'IS',
    address_lines: [
        'Háteigsvegur',
        '40',
        '\n'
    ],
    postal_code: '105',
    address_residential_indicator: undefined,
    city_locality: 'Reykjavík',
    company_name: 'Company Indridi ',
    email: 'indridi007@test.com',
    last_name: 'Bentsson',
    instructions: '',
    is_eu: false,
    name: 'Indridi  Bentsson',
    phone_number: '+3545971229',
    state_province: 'Höfuðborgarsvæði'
};
exports.IN_ADDRESS = {
    first_name: 'Mitul',
    country_code: 'IN',
    address_lines: [
        'Rd Number 8',
        '113',
        '\n'
    ],
    postal_code: '400062',
    address_residential_indicator: undefined,
    city_locality: 'Mumbai',
    company_name: 'Company Mitul',
    email: 'mitulek123@test.com',
    last_name: 'Dhawan',
    instructions: '',
    is_eu: false,
    name: 'Mitul Dhawan',
    phone_number: '+919869060305',
    state_province: 'Maharashtra'
};
exports.IE_ADDRESS = {
    first_name: 'Daire',
    country_code: 'IE',
    address_lines: [
        'Old Cratloe Rd,',
        '12',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Dublin',
    company_name: 'Company Daire',
    email: 'daire890@test.com',
    last_name: 'Allen',
    instructions: '',
    is_eu: false,
    name: 'Daire Allen',
    phone_number: '+35313432345',
    state_province: 'Leinster'
};
exports.IT_ADDRESS = {
    first_name: 'Lorenzo',
    country_code: 'IT',
    address_lines: [
        'Via del Podere Fiume',
        '4',
        '\n'
    ],
    postal_code: '00168',
    address_residential_indicator: undefined,
    city_locality: 'Roma',
    company_name: 'Company Lorenzo',
    email: 'LiberataLorenzo@test.com',
    last_name: 'Liberata',
    instructions: '',
    is_eu: false,
    name: 'Lorenzo Liberata',
    phone_number: '+390678982332',
    state_province: 'RM'
};
exports.JM_ADDRESS = {
    first_name: 'Amal',
    country_code: 'JM',
    address_lines: [
        'Blue Mahoe Place',
        '2',
        '\n'
    ],
    postal_code: 'JMBTS07',
    address_residential_indicator: undefined,
    city_locality: 'Lyssons',
    company_name: 'Company Amal',
    email: 'amal.hyatt@test.com',
    last_name: 'Hyatt',
    instructions: '',
    is_eu: false,
    name: 'Amal Hyatt',
    phone_number: '+1 876-478-0000',
    state_province: 'Surrey'
};
exports.JP_ADDRESS = {
    first_name: 'Shuichi',
    country_code: 'JP',
    address_lines: [
        'Nemuro-shi',
        '395-2',
        '\n'
    ],
    postal_code: '088-1781',
    address_residential_indicator: undefined,
    city_locality: 'Nemuro-shi',
    company_name: 'Company Shuichi',
    email: 'shuici.uchida@test.com',
    last_name: 'Uchida',
    instructions: '',
    is_eu: false,
    name: 'Shuichi Uchida',
    phone_number: '+81 153-27-0000',
    state_province: 'Hokkaido'
};
exports.KZ_ADDRESS = {
    first_name: 'Andrei',
    country_code: 'KZ',
    address_lines: [
        'Evrazia Avenue',
        '59',
        '1'
    ],
    postal_code: '090000',
    address_residential_indicator: undefined,
    city_locality: 'Uralsk',
    company_name: 'Company Andrei',
    email: 'andrei.gurman@test.com',
    last_name: 'Gurman',
    instructions: '',
    is_eu: false,
    name: 'Andrei Gurman',
    phone_number: '+7 705 390 3232',
    state_province: 'Batıs Qazaqstan'
};
exports.LV_ADDRESS = {
    first_name: 'Roberts',
    country_code: 'LV',
    address_lines: [
        'Lilijas iela',
        '14',
        '\n'
    ],
    postal_code: 'LV-1055',
    address_residential_indicator: undefined,
    city_locality: 'Riga',
    company_name: 'Company Roberts',
    email: 'robertus89@test.com',
    last_name: 'Šabala',
    instructions: '',
    is_eu: false,
    name: 'Roberts Šabala',
    phone_number: '+37180004535',
    state_province: 'Vidzeme'
};
exports.LI_ADDRESS = {
    first_name: 'Alina',
    country_code: 'LI',
    address_lines: [
        'Mitteldorf',
        '32',
        '\n'
    ],
    postal_code: '9490',
    address_residential_indicator: undefined,
    city_locality: 'Vaduz',
    company_name: 'Company Alina',
    email: 'sele123alina@test.com',
    last_name: 'Sele',
    instructions: '',
    is_eu: false,
    name: 'Alina Sele',
    phone_number: '+4235433311',
    state_province: 'Vaduz'
};
exports.LU_ADDRESS = {
    first_name: 'Eric',
    country_code: 'LU',
    address_lines: [
        'Boulevard Joseph II',
        '16',
        '\n'
    ],
    postal_code: '1840',
    address_residential_indicator: undefined,
    city_locality: 'Luxembourg',
    company_name: 'Company Eric',
    email: 'ericJ18@test.com',
    last_name: 'Joubert',
    instructions: '',
    is_eu: false,
    name: 'Eric Joubert',
    phone_number: '+35226792231',
    state_province: 'Luxembourg'
};
exports.MK_ADDRESS = {
    first_name: 'Dusana',
    country_code: 'MK',
    address_lines: [
        'Mitropolit Teodosij Gologanov',
        '45',
        '\n'
    ],
    postal_code: '1000',
    address_residential_indicator: undefined,
    city_locality: 'Skopje',
    company_name: 'Company Dusana',
    email: 'markovietz@test.com',
    last_name: 'Markovski',
    instructions: '',
    is_eu: false,
    name: 'Dusana Markovski',
    phone_number: '+38923121848',
    state_province: 'Skopje'
};
exports.MT_ADDRESS = {
    first_name: 'Alfred',
    country_code: 'MT',
    address_lines: [
        'Strait St',
        '103',
        '\n'
    ],
    postal_code: 'VLT 1436',
    address_residential_indicator: undefined,
    city_locality: 'Valetta',
    company_name: 'Company Alfred',
    email: 'hoogger@test.com',
    last_name: 'Hogg',
    instructions: '',
    is_eu: false,
    name: 'Alfred Hogg',
    phone_number: '+35621331591',
    state_province: 'South Eastern Region'
};
exports.MX_ADDRESS = {
    first_name: 'Hector',
    country_code: 'MX',
    address_lines: [
        'Ignacio Zaragoza',
        '10',
        '\n'
    ],
    postal_code: '77360',
    address_residential_indicator: undefined,
    city_locality: 'Ignacio Zaragoza',
    company_name: 'Company Hector',
    email: 'hector.ochoa@test.com',
    last_name: 'Ochoa',
    instructions: '',
    is_eu: false,
    name: 'Hector Ochoa',
    phone_number: '+52 998 307 0000',
    state_province: 'Quintana Roo '
};
exports.MD_ADDRESS = {
    first_name: 'Nelia',
    country_code: 'MD',
    address_lines: [
        'Suceava St',
        '110',
        '\n'
    ],
    postal_code: '2051',
    address_residential_indicator: undefined,
    city_locality: 'Chișinău',
    company_name: 'Company Nelia',
    email: 'turcanonelia09@test.com',
    last_name: 'Turcano',
    instructions: '',
    is_eu: false,
    name: 'Nelia Turcano',
    phone_number: '+37368105676',
    state_province: 'Chișinău'
};
exports.MC_ADDRESS = {
    first_name: 'Raphael',
    country_code: 'MC',
    address_lines: [
        'Allées des Boulingrins',
        '68',
        '\n'
    ],
    postal_code: '98000',
    address_residential_indicator: undefined,
    city_locality: 'Monaco',
    company_name: 'Company Raphael',
    email: 'raphael.sidibe@test.com',
    last_name: 'Sidibe',
    instructions: '',
    is_eu: false,
    name: 'Raphael Sidibe',
    phone_number: '+377 6 82 20 08 93',
    state_province: '\n'
};
exports.ME_ADDRESS = {
    first_name: 'Rajan',
    country_code: 'ME',
    address_lines: [
        'Blv Save Kovačevića',
        '74',
        '\n'
    ],
    postal_code: '81000',
    address_residential_indicator: undefined,
    city_locality: 'Podgorica',
    company_name: 'Company Rajan',
    email: 'mugosaraja1@test.com',
    last_name: 'Mugoša',
    instructions: '',
    is_eu: false,
    name: 'Rajan Mugoša',
    phone_number: '+38220632623',
    state_province: 'Podgorica'
};
exports.NL_ADDRESS = {
    first_name: 'Bouchra',
    country_code: 'NL',
    address_lines: [
        'Jacob Van Lennepkade',
        '215',
        '\n'
    ],
    postal_code: '1054 ZP',
    address_residential_indicator: undefined,
    city_locality: 'Amsterdam',
    company_name: 'Company Bouchra',
    email: 'Bouchra@test.com',
    last_name: 'van Welingenten',
    instructions: '',
    is_eu: false,
    name: 'Bouchra van Welingenten',
    phone_number: '+31207604000',
    state_province: 'Noord-Holland'
};
exports.NZ_ADDRESS = {
    first_name: 'Marco',
    country_code: 'NZ',
    address_lines: [
        'Waterfront Dr',
        '137',
        '\n'
    ],
    postal_code: '0420',
    address_residential_indicator: undefined,
    city_locality: 'Mangonui',
    company_name: 'Company Marco',
    email: 'marco.durante@test.com',
    last_name: 'Durante',
    instructions: '',
    is_eu: false,
    name: 'Marco Durante',
    phone_number: '+64 9-406 0000',
    state_province: 'Northland'
};
exports.NI_ADDRESS = {
    first_name: 'David',
    country_code: 'NI',
    address_lines: [
        'De la Iglesia Bautista',
        '1',
        '\n'
    ],
    postal_code: '35200',
    address_residential_indicator: undefined,
    city_locality: 'Palacagüina',
    company_name: 'Company David',
    email: 'david.barrera@test.com',
    last_name: 'Barrera',
    instructions: '',
    is_eu: false,
    name: 'David Barrera',
    phone_number: '+505 2722 0000',
    state_province: 'Madriz'
};
exports.NG_ADDRESS = {
    first_name: 'Adebamgbe',
    country_code: 'NG',
    address_lines: [
        'Aguiyi Ironsi Street',
        '1',
        '\n'
    ],
    postal_code: '900001',
    address_residential_indicator: undefined,
    city_locality: 'Abuja',
    company_name: 'Company Adebamgbe',
    email: '\n',
    last_name: 'Mohammed',
    instructions: '',
    is_eu: false,
    name: 'Adebamgbe Mohammed',
    phone_number: '+234 708 070 1003',
    state_province: '\n'
};
exports.NO_ADDRESS = {
    first_name: 'Arvid',
    country_code: 'NO',
    address_lines: [
        'Trondheimsveien',
        '37',
        '\n'
    ],
    postal_code: '0560',
    address_residential_indicator: undefined,
    city_locality: 'Oslo',
    company_name: 'Company Arvid',
    email: 'arvidm12@test.com',
    last_name: 'Mykland',
    instructions: '',
    is_eu: false,
    name: 'Arvid Mykland',
    phone_number: '+4722686130',
    state_province: 'Østlandet'
};
exports.PA_ADDRESS = {
    first_name: 'Romeo',
    country_code: 'PA',
    address_lines: [
        'Finca',
        '12',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Changuinola',
    company_name: 'Company Romeo',
    email: 'romeo.muraco@test.com',
    last_name: 'Muraco',
    instructions: '',
    is_eu: false,
    name: 'Romeo Muraco',
    phone_number: '+507 760-0000',
    state_province: 'Bocas del Toro'
};
exports.PK_ADDRESS = {
    first_name: 'Shabir',
    country_code: 'PK',
    address_lines: [
        '\n',
        '\n',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: '\n',
    company_name: 'Company Shabir',
    email: 'shabir.bilal@test.com',
    last_name: 'Bilal',
    instructions: '',
    is_eu: false,
    name: 'Shabir Bilal',
    phone_number: '\n',
    state_province: 'Baluchistan'
};
exports.PY_ADDRESS = {
    first_name: 'Sofia',
    country_code: 'PY',
    address_lines: [
        'Av Mariscal Francisco San López',
        '15',
        '\n'
    ],
    postal_code: '6360',
    address_residential_indicator: undefined,
    city_locality: 'Obligado',
    company_name: 'Company Sofia',
    email: 'testsofiatest@test.com',
    last_name: 'Villar',
    instructions: '',
    is_eu: false,
    name: 'Sofia Villar',
    phone_number: '+595221602162',
    state_province: 'Itapúa'
};
exports.PE_ADDRESS = {
    first_name: 'Alejandro',
    country_code: 'PE',
    address_lines: [
        'Calle Ricardo Palma',
        '159',
        '\n'
    ],
    postal_code: '16001',
    address_residential_indicator: undefined,
    city_locality: 'Iquitos ',
    company_name: 'Company Alejandro',
    email: '\n',
    last_name: 'Guarachi',
    instructions: '',
    is_eu: false,
    name: 'Alejandro Guarachi',
    phone_number: '+51 65 332332',
    state_province: 'Loreto'
};
exports.PT_ADDRESS = {
    first_name: 'Rui',
    country_code: 'PT',
    address_lines: [
        'R. do Diário de Notícias',
        '39',
        '\n'
    ],
    postal_code: '1200-141',
    address_residential_indicator: undefined,
    city_locality: 'Lisboa',
    company_name: 'Company Rui',
    email: 'rui.patricio@test.com',
    last_name: 'Patricio',
    instructions: '',
    is_eu: false,
    name: 'Rui Patricio',
    phone_number: '+351 961 339 696',
    state_province: 'Lisboa'
};
exports.PR_ADDRESS = {
    first_name: 'Antonio',
    country_code: 'PRI',
    address_lines: [
        'Calle Vendig',
        '2',
        '\n'
    ],
    postal_code: '00907',
    address_residential_indicator: undefined,
    city_locality: 'San Juan',
    company_name: 'Company Antonio',
    email: 'apaoli@test.com',
    last_name: 'Paoli',
    instructions: '',
    is_eu: false,
    name: 'Antonio Paoli',
    phone_number: '+1 787-724-6300',
    state_province: 'PR'
};
exports.QA_ADDRESS = {
    first_name: 'Saad',
    country_code: 'QA',
    address_lines: [
        'Al Haya',
        '51',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Umm Jurn',
    company_name: 'Company Saad',
    email: 'saad.bakri@test.com',
    last_name: 'Al Bakri',
    instructions: '',
    is_eu: false,
    name: 'Saad Al Bakri',
    phone_number: '+974 7007 0000',
    state_province: 'Ad-Da’ajin'
};
exports.KR_ADDRESS = {
    first_name: 'Jae-Sang',
    country_code: 'KR',
    address_lines: [
        'Cheongdam-dong',
        '37-12',
        '\n'
    ],
    postal_code: '06065',
    address_residential_indicator: undefined,
    city_locality: 'Seoul',
    company_name: 'Company Jae-Sang',
    email: 'gstyle@test.com',
    last_name: 'Park',
    instructions: '',
    is_eu: false,
    name: 'Jae-Sang Park',
    phone_number: '+82 2-547-1477',
    state_province: '\n'
};
exports.RO_ADDRESS = {
    first_name: 'Costel',
    country_code: 'RO',
    address_lines: [
        'Strada 1 Decembrie',
        '2',
        '\n'
    ],
    postal_code: '827005',
    address_residential_indicator: undefined,
    city_locality: 'Baia',
    company_name: 'Company Costel',
    email: 'costel.lobont@test.com',
    last_name: 'Lobont',
    instructions: '',
    is_eu: false,
    name: 'Costel Lobont',
    phone_number: '+40 240 564 138',
    state_province: 'Tulcea'
};
exports.RU_ADDRESS = {
    first_name: 'Igor',
    country_code: 'RU',
    address_lines: [
        'Nekrasova',
        '3',
        '\n'
    ],
    postal_code: '180000',
    address_residential_indicator: undefined,
    city_locality: 'Pskov',
    company_name: 'Company Igor',
    email: 'igor.shunin@test.com',
    last_name: 'Shunin',
    instructions: '',
    is_eu: false,
    name: 'Igor Shunin',
    phone_number: '+7 811 279-00-00',
    state_province: "Severo-Zapadnyy federal'nyy okrug"
};
exports.LC_ADDRESS = {
    first_name: 'Eden',
    country_code: 'LC',
    address_lines: [
        'John Compton Hwy',
        '57',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Castries',
    company_name: 'Company Eden',
    email: 'eden.henry@test.com',
    last_name: 'Henry',
    instructions: '',
    is_eu: false,
    name: 'Eden Henry',
    phone_number: '+1 758-452-0000',
    state_province: 'Castries'
};
exports.SV_ADDRESS = {
    first_name: 'Alfredo',
    country_code: 'SV',
    address_lines: [
        'Calle Poniente',
        '8',
        '\n'
    ],
    postal_code: '2321',
    address_residential_indicator: undefined,
    city_locality: 'San Salvador',
    company_name: 'Company Alfredo',
    email: 'alfredo.alas@test.com',
    last_name: 'Alas',
    instructions: '',
    is_eu: false,
    name: 'Alfredo Alas',
    phone_number: '+503 2536 0000',
    state_province: 'San Salvador'
};
exports.SM_ADDRESS = {
    first_name: 'Andy',
    country_code: 'SM',
    address_lines: [
        'Via Basilicius',
        '11',
        '\n'
    ],
    postal_code: '47890',
    address_residential_indicator: undefined,
    city_locality: 'Città di San Marino',
    company_name: 'Company Andy',
    email: 'basiciluss@test.com',
    last_name: 'Gennari',
    instructions: '',
    is_eu: false,
    name: 'Andy Gennari',
    phone_number: '+3780549991886',
    state_province: 'Castello della Città di San Marino'
};
exports.RS_ADDRESS = {
    first_name: 'Stefan',
    country_code: 'RS',
    address_lines: [
        'Ledi Pedžet',
        '2',
        '\n'
    ],
    postal_code: '11040',
    address_residential_indicator: undefined,
    city_locality: 'Beograd',
    company_name: 'Company Stefan',
    email: 'stefanik1@test.com',
    last_name: 'Anđić',
    instructions: '',
    is_eu: false,
    name: 'Stefan Anđić',
    phone_number: '+381112630625',
    state_province: 'Beograd'
};
exports.SK_ADDRESS = {
    first_name: 'Jan',
    country_code: 'SK',
    address_lines: [
        'Partizánska',
        '469',
        '63'
    ],
    postal_code: '962 12',
    address_residential_indicator: undefined,
    city_locality: 'Detva',
    company_name: 'Company Jan',
    email: 'jan.kuciak@test.com',
    last_name: 'Kuciak',
    instructions: '',
    is_eu: false,
    name: 'Jan Kuciak',
    phone_number: '+421 45/545 49 74',
    state_province: 'Banskobystrický '
};
exports.SI_ADDRESS = {
    first_name: 'Jan',
    country_code: 'SI',
    address_lines: [
        'Krakovski nasip',
        '12',
        '\n'
    ],
    postal_code: '1000',
    address_residential_indicator: undefined,
    city_locality: 'Ljubljana',
    company_name: 'Company Jan',
    email: 'jan.handanovic@test.com',
    last_name: 'Handanovic',
    instructions: '',
    is_eu: false,
    name: 'Jan Handanovic',
    phone_number: '+386 40 831 840',
    state_province: '\n'
};
exports.ES_ADDRESS = {
    first_name: 'Sergio',
    country_code: 'ES',
    address_lines: [
        'Avinguda de Rio de Janeiro',
        '42',
        '\n'
    ],
    postal_code: '08016',
    address_residential_indicator: undefined,
    city_locality: 'Barcelona',
    company_name: 'Company Sergio',
    email: 'sergio.rajoy@test.com',
    last_name: 'Rajoy',
    instructions: '',
    is_eu: false,
    name: 'Sergio Rajoy',
    phone_number: '+34 932 76 50 70',
    state_province: 'Barcelona'
};
exports.SR_ADDRESS = {
    first_name: 'Shivani',
    country_code: 'SR',
    address_lines: [
        'Jaggernath Lachmon St',
        '167',
        '\n'
    ],
    postal_code: 'Paramaribo',
    address_residential_indicator: undefined,
    city_locality: 'Paramaribo',
    company_name: 'Company Shivani',
    email: 'grosdma@test.com',
    last_name: 'Kwasie',
    instructions: '',
    is_eu: false,
    name: 'Shivani Kwasie',
    phone_number: '+5974624401',
    state_province: 'Paramaribo'
};
exports.SE_ADDRESS = {
    first_name: 'Rasmus',
    country_code: 'SE',
    address_lines: [
        'Föreningsgatan',
        '6',
        '\n'
    ],
    postal_code: '98131',
    address_residential_indicator: undefined,
    city_locality: 'Kiruna',
    company_name: 'Company Rasmus',
    email: 'rasmus.rasmusen@test.com',
    last_name: 'Rasmusen',
    instructions: '',
    is_eu: false,
    name: 'Rasmus Rasmusen',
    phone_number: '+46 980 155 00',
    state_province: 'Lappland'
};
exports.CH_ADDRESS = {
    first_name: 'Yann',
    country_code: 'CH',
    address_lines: [
        'Dorf',
        '16',
        '\n'
    ],
    postal_code: '9057',
    address_residential_indicator: undefined,
    city_locality: 'Weissbad',
    company_name: 'Company Yann',
    email: 'yann..mvogo@test.com',
    last_name: 'Mvogo',
    instructions: '',
    is_eu: false,
    name: 'Yann Mvogo',
    phone_number: '+41 71 797 02 40',
    state_province: 'Appenzell Innerrhoden'
};
exports.TR_ADDRESS = {
    first_name: 'Bedri',
    country_code: 'TR',
    address_lines: [
        'Sokullu Mehmet Paşa Cd.',
        '184',
        '\n'
    ],
    postal_code: '06000',
    address_residential_indicator: undefined,
    city_locality: 'Ankara',
    company_name: 'Company Bedri',
    email: 'tuekybedri@test.com',
    last_name: 'Mandıralı',
    instructions: '',
    is_eu: false,
    name: 'Bedri Mandıralı',
    phone_number: '+903123111231',
    state_province: 'Ankara'
};
exports.UA_ADDRESS = {
    first_name: 'Andryi',
    country_code: 'UA',
    address_lines: [
        'Vulytsya Tsentralʹna',
        '23A',
        '\n'
    ],
    postal_code: '93500',
    address_residential_indicator: undefined,
    city_locality: 'Novoaidar',
    company_name: 'Company Andryi',
    email: 'andryi.boyko@test.com',
    last_name: 'Boyko',
    instructions: '',
    is_eu: false,
    name: 'Andryi Boyko',
    phone_number: '+380 6445 94258',
    state_province: 'Łuhanśka Oblast'
};
exports.AE_ADDRESS = {
    first_name: 'Walid',
    country_code: 'AE',
    address_lines: [
        'Al Bateen',
        '12',
        '\n'
    ],
    postal_code: '\n',
    address_residential_indicator: undefined,
    city_locality: 'Abu Zabi',
    company_name: 'Company Walid',
    email: 'walid.saleh@test.com',
    last_name: 'Saleh',
    instructions: '',
    is_eu: false,
    name: 'Walid Saleh',
    phone_number: '+971 800 555',
    state_province: 'Abu Zabi'
};
exports.US_ADDRESS = {
    first_name: 'Petyr',
    country_code: 'US',
    address_lines: [
        'Boston Road',
        '1600',
        '\n'
    ],
    postal_code: '10460',
    address_residential_indicator: undefined,
    city_locality: 'New York',
    company_name: 'Company Petyr',
    email: 'petyr.baelish@test.com',
    last_name: 'Baelish',
    instructions: '',
    is_eu: false,
    name: 'Petyr Baelish',
    phone_number: '+1 718-542-7914',
    state_province: 'NY'
};
exports.UY_ADDRESS = {
    first_name: 'Fernando',
    country_code: 'UY',
    address_lines: [
        'Continuación Cub',
        '4384',
        '\n'
    ],
    postal_code: '12800 ',
    address_residential_indicator: undefined,
    city_locality: 'Montevideo',
    company_name: 'Company Fernando',
    email: 'fernando.silva@test.com',
    last_name: 'Silva',
    instructions: '',
    is_eu: false,
    name: 'Fernando Silva',
    phone_number: '+598 97 055 000',
    state_province: 'Montevideo'
};
exports.VA_ADDRESS = {
    first_name: 'Francisco',
    country_code: 'VA',
    address_lines: [
        'Via della Posta',
        '12',
        '\n'
    ],
    postal_code: '00120',
    address_residential_indicator: undefined,
    city_locality: 'Città del Vaticano',
    company_name: 'Company Francisco',
    email: 'francisco.sacro@test.com',
    last_name: 'Sacro',
    instructions: '',
    is_eu: false,
    name: 'Francisco Sacro',
    phone_number: '+39 06 6988 0000',
    state_province: 'Aragua'
};
exports.VE_ADDRESS = {
    first_name: 'Daniel',
    country_code: 'VE',
    address_lines: [
        'Calle Providencia',
        '12',
        '\n'
    ],
    postal_code: '2122',
    address_residential_indicator: undefined,
    city_locality: 'Cagua',
    company_name: 'Company Daniel',
    email: 'daniel.baroja@test.com',
    last_name: 'Baroja',
    instructions: '',
    is_eu: false,
    name: 'Daniel Baroja',
    phone_number: '+58 244-3961723',
    state_province: 'Aragua'
};
exports.AD_ADDRESS = {
    first_name: 'Amanda',
    country_code: 'AD',
    address_lines: [
        'Carrer Prat de la Creu',
        '88',
        '\n'
    ],
    postal_code: 'AD500',
    address_residential_indicator: undefined,
    city_locality: 'Andorra la Vella',
    company_name: 'Company Amanda',
    email: 'son8890@test.com',
    last_name: 'Sonejee',
    instructions: '',
    is_eu: false,
    name: 'Amanda Sonejee',
    phone_number: '+376874744',
    state_province: 'Escaldes-Engordany'
};
exports.GB_ADDRESS = {
    country_code: 'GB',
    address_lines: ['London Bridge St 32', 'N End 42'],
    postal_code: 'SE1 9SG',
    address_residential_indicator: connect_carrier_api_1.AddressResidentialIndicator.Yes,
    city_locality: 'London',
    company_name: 'Smith Corp.',
    email: 'smnith@test.com',
    first_name: 'John',
    last_name: 'Smith',
    instructions: '',
    is_eu: false,
    name: 'John Smith',
    phone_number: '+442087956001',
    state_province: 'Londonderry'
};
exports.TESTS_ADDRESS_DATA_SOURCE = new Map([
    [countries_1.Countries.DEFAULT, exports.DEFAULT_ADDRESS],
    [countries_1.Countries.PL, exports.PL_ADDRESS],
    [countries_1.Countries.GB, exports.GB_ADDRESS],
    [countries_1.Countries.AL, exports.AL_ADDRESS],
    [countries_1.Countries.AR, exports.AR_ADDRESS],
    [countries_1.Countries.AU, exports.AU_ADDRESS],
    [countries_1.Countries.AM, exports.AM_ADDRESS],
    [countries_1.Countries.AG, exports.AG_ADDRESS],
    [countries_1.Countries.AT, exports.AT_ADDRESS],
    [countries_1.Countries.BB, exports.BB_ADDRESS],
    [countries_1.Countries.BY, exports.BY_ADDRESS],
    [countries_1.Countries.BE, exports.BE_ADDRESS],
    [countries_1.Countries.BZ, exports.BZ_ADDRESS],
    [countries_1.Countries.BO, exports.BO_ADDRESS],
    [countries_1.Countries.BA, exports.BA_ADDRESS],
    [countries_1.Countries.BR, exports.BR_ADDRESS],
    [countries_1.Countries.BG, exports.BG_ADDRESS],
    [countries_1.Countries.CA, exports.CA_ADDRESS],
    [countries_1.Countries.CL, exports.CL_ADDRESS],
    [countries_1.Countries.CN, exports.CN_ADDRESS],
    [countries_1.Countries.CO, exports.CO_ADDRESS],
    [countries_1.Countries.HR, exports.HR_ADDRESS],
    [countries_1.Countries.CU, exports.CU_ADDRESS],
    [countries_1.Countries.CY, exports.CY_ADDRESS],
    [countries_1.Countries.CZ, exports.CZ_ADDRESS],
    [countries_1.Countries.DK, exports.DK_ADDRESS],
    [countries_1.Countries.EC, exports.EC_ADDRESS],
    [countries_1.Countries.EE, exports.EE_ADDRESS],
    [countries_1.Countries.FO, exports.FO_ADDRESS],
    [countries_1.Countries.FI, exports.FI_ADDRESS],
    [countries_1.Countries.FR, exports.FR_ADDRESS],
    [countries_1.Countries.GE, exports.GE_ADDRESS],
    [countries_1.Countries.GF, exports.GF_ADDRESS],
    [countries_1.Countries.DE, exports.DE_ADDRESS],
    [countries_1.Countries.GI, exports.GI_ADDRESS],
    [countries_1.Countries.GR, exports.GR_ADDRESS],
    [countries_1.Countries.GT, exports.GT_ADDRESS],
    [countries_1.Countries.GY, exports.GY_ADDRESS],
    [countries_1.Countries.HT, exports.HT_ADDRESS],
    [countries_1.Countries.HN, exports.HN_ADDRESS],
    [countries_1.Countries.HU, exports.HU_ADDRESS],
    [countries_1.Countries.IS, exports.IS_ADDRESS],
    [countries_1.Countries.IN, exports.IN_ADDRESS],
    [countries_1.Countries.IE, exports.IE_ADDRESS],
    [countries_1.Countries.IT, exports.IT_ADDRESS],
    [countries_1.Countries.JM, exports.JM_ADDRESS],
    [countries_1.Countries.JP, exports.JP_ADDRESS],
    [countries_1.Countries.KZ, exports.KZ_ADDRESS],
    [countries_1.Countries.LV, exports.LV_ADDRESS],
    [countries_1.Countries.LI, exports.LI_ADDRESS],
    [countries_1.Countries.LU, exports.LU_ADDRESS],
    [countries_1.Countries.MK, exports.MK_ADDRESS],
    [countries_1.Countries.MT, exports.MT_ADDRESS],
    [countries_1.Countries.MX, exports.MX_ADDRESS],
    [countries_1.Countries.MD, exports.MD_ADDRESS],
    [countries_1.Countries.MC, exports.MC_ADDRESS],
    [countries_1.Countries.ME, exports.ME_ADDRESS],
    [countries_1.Countries.NL, exports.NL_ADDRESS],
    [countries_1.Countries.NZ, exports.NZ_ADDRESS],
    [countries_1.Countries.NI, exports.NI_ADDRESS],
    [countries_1.Countries.NG, exports.NG_ADDRESS],
    [countries_1.Countries.NO, exports.NO_ADDRESS],
    [countries_1.Countries.PA, exports.PA_ADDRESS],
    [countries_1.Countries.PK, exports.PK_ADDRESS],
    [countries_1.Countries.PY, exports.PY_ADDRESS],
    [countries_1.Countries.PE, exports.PE_ADDRESS],
    [countries_1.Countries.PT, exports.PT_ADDRESS],
    [countries_1.Countries.PR, exports.PR_ADDRESS],
    [countries_1.Countries.QA, exports.QA_ADDRESS],
    [countries_1.Countries.KR, exports.KR_ADDRESS],
    [countries_1.Countries.RO, exports.RO_ADDRESS],
    [countries_1.Countries.RU, exports.RU_ADDRESS],
    [countries_1.Countries.LC, exports.LC_ADDRESS],
    [countries_1.Countries.SV, exports.SV_ADDRESS],
    [countries_1.Countries.SM, exports.SM_ADDRESS],
    [countries_1.Countries.RS, exports.RS_ADDRESS],
    [countries_1.Countries.SK, exports.SK_ADDRESS],
    [countries_1.Countries.SI, exports.SI_ADDRESS],
    [countries_1.Countries.ES, exports.ES_ADDRESS],
    [countries_1.Countries.SR, exports.SR_ADDRESS],
    [countries_1.Countries.SE, exports.SE_ADDRESS],
    [countries_1.Countries.CH, exports.CH_ADDRESS],
    [countries_1.Countries.TR, exports.TR_ADDRESS],
    [countries_1.Countries.UA, exports.UA_ADDRESS],
    [countries_1.Countries.AE, exports.AE_ADDRESS],
    [countries_1.Countries.US, exports.US_ADDRESS],
    [countries_1.Countries.UY, exports.UY_ADDRESS],
    [countries_1.Countries.VA, exports.VA_ADDRESS],
    [countries_1.Countries.VE, exports.VE_ADDRESS],
    [countries_1.Countries.AD, exports.AD_ADDRESS]
]);
