// @shipengine/connect-carrier-api/src/models/importer-records/importer-of-record.ts
import { TaxIdentifier } from '../taxes/tax-identifier';

/** @description Basic structure for a Importer Records */
export class ImporterOfRecord {
  /** @description Tax Ids associated with the party responsible for ensuring that imported goods comply with all customs and legal requirements of the country of import (importer of record) */
  tax_identifiers?: TaxIdentifier[];
  /** @description Indicates Name of the owner or purchaser of the products being imported into a destination country */
  name?: string;
  /** @description Indicates Company Name of the owner or purchaser of the products being imported into a destination country*/
  company_name?: string;
  /** @description Indicates address line one of the owner or purchaser of the products being imported into a destination country*/
  address_line1?: string;
  /** @description Indicates address  of the owner or purchaser of the products being imported into a destination country*/
  address_line2?: string;
  /** @description Indicates City  or Locality  of the owner or purchaser of the products being imported into a destination country */
  city_locality?: string;
  /** @description Indicates statate or providence  of the owner or purchaser of the products being imported into a destination country */
  state_province?: string;
  /** @description : Indicates postal code of the owner or purchaser of the products being imported into a destination country*/
  postal_code?: string;
  /** @description : Indicates country code of the owner or purchaser of the products being imported into a destination country*/
  country_code?: string;
  /** @description : Indicates phone number of the owner or purchaser of the products being imported into a destination country*/
  phone?: string;
  /** @description : Email associated with the party responsible for ensuring that imported goods comply with all customs and legal requirements of the country of import (importer of record)*/
  email?: string;
}
