// @shipengine/connect/lib/commands/pack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const base_command_1 = tslib_1.__importDefault(require("../base-command"));
const command_1 = require("@oclif/command");
const package_app_1 = require("../core/package-app");
class Pack extends base_command_1.default {
    static description = 'Package your app';
    static examples = ['$ connect pack'];
    // TODO: come up with a convention for turning off spinners if the user desires
    // TODO: implement a quiet command?
    static flags = {
        help: command_1.flags.help({
            char: 'h',
            description: 'Show help for the pack command',
        }),
    };
    async run() {
        // When the -h flag is present the following line haults execution
        this.parse(Pack);
        try {
            await (0, package_app_1.packageApp)();
        }
        catch (error) {
            if ((0, package_app_1.isAppFailedToPackageError)(error)) {
                return this.error(error.message, {
                    exit: 1,
                });
            }
            throw error;
        }
    }
}
exports.default = Pack;
