// @shipengine/connect/lib/commands/whoami.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const base_command_1 = tslib_1.__importDefault(require("../base-command"));
const command_1 = require("@oclif/command");
const unauthenticated_error_1 = require("../core/unauthenticated-error");
class Whoami extends base_command_1.default {
    static description = 'Display the current logged in user';
    static aliases = ['whoami'];
    static flags = {
        help: command_1.flags.help({
            char: 'h',
            description: 'Show help for the whoami command',
        }),
        debug: command_1.flags.boolean({
            char: 'd',
            description: 'Show network debugging information',
            default: false,
            hidden: true,
        }),
    };
    async run() {
        // When the -h flag is present the following line haults execution
        const { flags } = this.parse(Whoami);
        try {
            const appUser = await this.getCurrentUser(flags.debug);
            this.log(`You are currently logged in as: ${appUser.name}`);
        }
        catch (error) {
            if (error instanceof unauthenticated_error_1.UnauthenticatedError) {
                return this.error('You are not currently logged in', {
                    exit: 1,
                });
            }
            else {
                return this.error(error, {
                    exit: 1,
                });
            }
        }
    }
}
exports.default = Whoami;
