// @shipengine/connect/lib/core/utils/log-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logStep = exports.logSkipStep = exports.logSkip = exports.logResults = exports.logPass = exports.logObject = exports.logFail = exports.log = exports.indentLines = exports.indent = void 0;
const tslib_1 = require("tslib");
const chalk_1 = tslib_1.__importDefault(require("chalk"));
function log(logLine) {
    console.log(logLine);
}
exports.log = log;
function repeat(str, n) {
    return new Array(n).join(str);
}
function indent(n) {
    return repeat('    ', n);
}
exports.indent = indent;
function indentLines(str, n) {
    return indent(n) + str.replace(/\n/g, `\n${indent(n)}`);
}
exports.indentLines = indentLines;
function logPass(logLine) {
    log(`${indent(2)}${chalk_1.default.bgGreen.black(' PASS ')} ${chalk_1.default.green(logLine)}`);
}
exports.logPass = logPass;
function logFail(logLine, withTag = true) {
    if (withTag) {
        log(`${indent(2)}${chalk_1.default.bgRed.black(' FAIL ')} ${chalk_1.default.red(logLine)}`);
    }
    else {
        log(`${indent(2)}${chalk_1.default.red(logLine)}`);
    }
}
exports.logFail = logFail;
function logSkip(logLine) {
    log(`${indent(2)}${chalk_1.default.bgWhite.black(' SKIP ')} ${chalk_1.default.gray(logLine)}`);
}
exports.logSkip = logSkip;
function logSkipStep(logLine) {
    log(chalk_1.default.white(logLine));
}
exports.logSkipStep = logSkipStep;
function logStep(logLine) {
    log(chalk_1.default.yellow(logLine));
}
exports.logStep = logStep;
function logObject(obj) {
    const jsonString = JSON.stringify([obj], null, 4).replace('[', '');
    console.log(jsonString.substring(0, jsonString.lastIndexOf(']')) +
        '' +
        jsonString.substring(jsonString.lastIndexOf(']') + 1));
}
exports.logObject = logObject;
function logResults(results) {
    log(`${chalk_1.default.green(results.passed.toString() + ' passing')} , ${chalk_1.default.red(results.failed.toString() + ' failing')}, ${results.skipped.toString() + ' skipped'}`);
}
exports.logResults = logResults;
