// connect-carrier-api-utils/lib/helpers/country/country-enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EuropeanUnionCountries = void 0;
var EuropeanUnionCountries;
(function (EuropeanUnionCountries) {
    EuropeanUnionCountries["AT"] = "AT";
    EuropeanUnionCountries["BE"] = "BE";
    EuropeanUnionCountries["BG"] = "BG";
    EuropeanUnionCountries["HR"] = "HR";
    EuropeanUnionCountries["CY"] = "CY";
    EuropeanUnionCountries["CZ"] = "CZ";
    EuropeanUnionCountries["DK"] = "DK";
    EuropeanUnionCountries["EE"] = "EE";
    EuropeanUnionCountries["FI"] = "FI";
    EuropeanUnionCountries["FR"] = "FR";
    EuropeanUnionCountries["DE"] = "DE";
    EuropeanUnionCountries["GR"] = "GR";
    EuropeanUnionCountries["HU"] = "HU";
    EuropeanUnionCountries["IE"] = "IE";
    EuropeanUnionCountries["IT"] = "IT";
    EuropeanUnionCountries["LV"] = "LV";
    EuropeanUnionCountries["LT"] = "LT";
    EuropeanUnionCountries["LU"] = "LU";
    EuropeanUnionCountries["MT"] = "MT";
    EuropeanUnionCountries["NL"] = "NL";
    EuropeanUnionCountries["PL"] = "PL";
    EuropeanUnionCountries["PT"] = "PT";
    EuropeanUnionCountries["RO"] = "RO";
    EuropeanUnionCountries["SK"] = "SK";
    EuropeanUnionCountries["SI"] = "SI";
    EuropeanUnionCountries["ES"] = "ES";
    EuropeanUnionCountries["SE"] = "SE";
})(EuropeanUnionCountries = exports.EuropeanUnionCountries || (exports.EuropeanUnionCountries = {}));
