// @shipengine/connect-carrier-api/lib/models/confirmation-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfirmationTypes = void 0;
/** @description Types of delivery confirmation */
var ConfirmationTypes;
(function (ConfirmationTypes) {
    /** @description No confirmation is requested */
    ConfirmationTypes["None"] = "None";
    /** @description Delivery confirmation is requested */
    ConfirmationTypes["Delivery"] = "Delivery";
    /** @description A signature is required. This signature may be a neighbor, building manager, or the recipient can authorize the release of the package (without being present) */
    ConfirmationTypes["Signature"] = "Signature";
    /** @description A signature of an adult is required */
    ConfirmationTypes["AdultSignature"] = "AdultSignature";
    /** @description Only supported by FedEx. The signature of somebody at the address is required */
    ConfirmationTypes["DirectSignature"] = "DirectSignature";
})(ConfirmationTypes || (exports.ConfirmationTypes = ConfirmationTypes = {}));
