// @shipengine/connect/lib/core/api-client/resources/sellers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Sellers {
    client;
    constructor(apiClient) {
        this.client = apiClient;
    }
    /**
     * Gets the sellers for an application scoped to the given API key.
     * @returns {Promise<Seller[]>} Promise that resolves to an Seller.
     */
    async getSellersForAppId(appId) {
        const response = await this.client.call({
            endpoint: `apps/${appId}/sellers`,
            method: 'GET',
        });
        return response;
    }
    /**
     * Creates a seller for an application scoped to the given API key.
     * @returns {Promise<Seller>} Promise that resolves to an Seller.
     */
    async createSeller(appId, email, password, country) {
        const response = await this.client.call({
            endpoint: `apps/${appId}/sellers`,
            method: 'POST',
            body: {
                email: email,
                password: password,
                firstName: 'Test',
                lastName: 'User',
                countryCode: country,
                type: 'shipstation',
            },
        });
        return response;
    }
}
exports.default = Sellers;
