// @shipengine/connect-carrier-api/lib/models/pickup/shipped-package-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShippedPackageSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const identifier_schema_1 = require("../identifier-schema");
const units_1 = require("../units");
const currency_schema_1 = require("../currency-schema");
const dimension_details_schema_1 = require("../units/dimension-details-schema");
const weight_details_schema_1 = require("../units/weight-details-schema");
const label_message_schema_1 = require("../labels/label-message-schema");
const customs_schema_1 = require("../customs/customs-schema");
exports.ShippedPackageSchema = joi_1.default.object({
    tracking_number: joi_1.default.string().optional(),
    tracking_url: joi_1.default.string().optional(),
    alternative_identifiers: joi_1.default.array().items(identifier_schema_1.IdentifierSchema).optional(),
    dimension_details: dimension_details_schema_1.DimensionDetailsSchema.optional(),
    weight_details: weight_details_schema_1.WeightDetailsSchema.optional(),
    package_code: joi_1.default.string().optional(),
    weight: joi_1.default.number().optional(),
    dimensions: units_1.DimensionsSchema.optional(),
    insured_value: currency_schema_1.CurrencySchema.required(),
    label_messages: label_message_schema_1.LabelMessageSchema.optional(),
    customs: customs_schema_1.CustomsSchema.optional(),
    products: joi_1.default.array().items(identifier_schema_1.IdentifierSchema).optional(),
});
