// @shipengine/connect/lib/app-servers/native-rating/create-dynamic-carrier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const vm2_1 = require("vm2");
/** Ensure that the carrier module has the correct shape
 * @param module Object into which code should be exported
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const validateCode = (module) => {
    if (module?.exports?.default === undefined) {
        return {
            errors: ['Code must have a default export'],
            hasGetZonesFunction: false,
        };
    }
    const errors = [];
    const { rateShipments, getZones } = module.exports.default;
    const hasGetZonesFunction = getZones !== undefined;
    if (typeof rateShipments !== 'function') {
        errors.push("Code must contain a function named 'rateShipments'");
    }
    else if (rateShipments.length !== 2) {
        errors.push("'rateShipments' function must have two parameters");
    }
    return {
        errors,
        hasGetZonesFunction,
    };
};
/** Create a dynamic carrier
 * @param code Code that should be used for the carrier
 * @returns Carrier with code ready to be executed
 */
const createDynamicCarrier = (code) => {
    // This is to make sure that the code we need to run doesn't get excluded because of dangling comments, etc.
    const resetCode = code + ';\n /* */ \n';
    const validationResults = new vm2_1.VM({
        sandbox: { module: {}, validateCode },
    }).run(`${resetCode}; validateCode(module);`);
    if (validationResults.errors?.length > 0) {
        throw new Error(validationResults.errors.join('; '));
    }
    const rateShipmentsScript = new vm2_1.VMScript(`${resetCode} module.exports.default.rateShipments(context, shipment);`);
    const executionVM = new vm2_1.VM({
        sandbox: {
            module: {},
            context: {},
            shipment: {},
            zoneContext: {},
            origin: {},
            destination: {},
        },
    });
    const carrier = {
        rateShipments: (context, shipment) => {
            return executionVM.setGlobals({ context, shipment }).run(rateShipmentsScript);
        },
    };
    return carrier;
};
exports.default = createDynamicCarrier;
