// @shipengine/connect-carrier-api/lib/models/tracking/track-event-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackEventSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const lat_long_schema_1 = require("./lat-long-schema");
const standardized_status_codes_schema_1 = require("./standardized-status-codes-schema");
exports.TrackEventSchema = joi_1.default.object({
    event_datetime: joi_1.default.string().optional().empty().isoDate(),
    event_datetime_local: joi_1.default.string().optional().empty().isoDate(),
    event_code: joi_1.default.string().optional().empty(),
    description: joi_1.default.string().optional().empty(),
    city: joi_1.default.string().optional().empty(),
    state: joi_1.default.string().optional().empty(),
    postal_code: joi_1.default.string().optional().empty(),
    country: joi_1.default.string().optional().empty(),
    company: joi_1.default.string().optional().empty(),
    signer: joi_1.default.string().optional().empty(),
    geo: lat_long_schema_1.LatLongSchema.optional(),
    status_code: standardized_status_codes_schema_1.StandardizedStatusCodesSchema.optional(),
});
