// @shipengine/connect-carrier-api/lib/models/tracking/tracking-info-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackingInfoSchema = void 0;
const tslib_1 = require("tslib");
const standardized_status_codes_schema_1 = require("./standardized-status-codes-schema");
const dimensions_schema_1 = require("../units/dimensions-schema");
const service_schema_1 = require("./service-schema");
const track_event_schema_1 = require("./track-event-schema");
const joi_1 = tslib_1.__importDefault(require("joi"));
exports.TrackingInfoSchema = joi_1.default.object({
    carrier_name: joi_1.default.string().optional().empty(),
    tracking_number: joi_1.default.string().optional().empty(),
    standardized_status_code: standardized_status_codes_schema_1.StandardizedStatusCodesSchema.required(),
    carrier_status_code: joi_1.default.string().optional().empty(),
    carrier_status_description: joi_1.default.string().optional().empty(),
    shipped_datetime: joi_1.default.string().optional().empty().isoDate(),
    estimated_delivery_datetime: joi_1.default.string().optional().empty(),
    actual_delivery_datetime: joi_1.default.string().optional().empty().isoDate(),
    shipping_problem_description: joi_1.default.string().optional().empty(),
    weight: joi_1.default.number().optional().empty(),
    dimensions: dimensions_schema_1.DimensionsSchema.optional(),
    service: service_schema_1.ServiceSchema.optional(),
    packaging: joi_1.default.string().optional().empty(),
    package_count: joi_1.default.number().optional().empty(),
    events: joi_1.default.array().optional().items(track_event_schema_1.TrackEventSchema),
    shipping_problem: joi_1.default.boolean().optional().empty(),
    shipping_problem_code: joi_1.default.string().optional().empty(),
    error_description: joi_1.default.string().optional().empty(),
});
