// @shipengine/connect-carrier-api/lib/app/metadata/confirmation-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfirmationTypeSchema = exports.ConfirmationTypeEnumSchema = exports.ConfirmationTypeEnum = exports.ConfirmationDictionarySchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
exports.ConfirmationDictionarySchema = joi_1.default.object({
    None: joi_1.default.string().optional(),
    Delivery: joi_1.default.string().optional(),
    Signature: joi_1.default.string().optional(),
    AdultSignature: joi_1.default.string().optional(),
    DirectSignature: joi_1.default.string().optional(),
});
var ConfirmationTypeEnum;
(function (ConfirmationTypeEnum) {
    ConfirmationTypeEnum["None"] = "None";
    ConfirmationTypeEnum["Delivery"] = "Delivery";
    ConfirmationTypeEnum["Signature"] = "Signature";
    ConfirmationTypeEnum["AdultSignature"] = "AdultSignature";
    ConfirmationTypeEnum["DirectSignature"] = "DirectSignature";
})(ConfirmationTypeEnum || (exports.ConfirmationTypeEnum = ConfirmationTypeEnum = {}));
exports.ConfirmationTypeEnumSchema = joi_1.default.string().valid(...Object.values(ConfirmationTypeEnum));
exports.ConfirmationTypeSchema = joi_1.default.object({
    Name: joi_1.default.string().optional().max(50),
    Type: exports.ConfirmationTypeEnumSchema.optional(),
});
