// @shipengine/connect-carrier-api/lib/app/metadata/connection-names-configuration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionNamesConfigurationSchema = exports.MapOfObjectsSchema = exports.ConnectionNamesSchema = exports.ConnectionNamesConfiguration = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
class ConnectionNamesConfiguration {
    /** @description A unique list of connection names that the integration supports. */
    connection_names;
    /** @description The connection name that will be used when none is selected. */
    default_connection_name;
}
exports.ConnectionNamesConfiguration = ConnectionNamesConfiguration;
exports.ConnectionNamesSchema = joi_1.default.alternatives().try('sandbox', 'production');
exports.MapOfObjectsSchema = joi_1.default.object().pattern(exports.ConnectionNamesSchema, joi_1.default.object());
exports.ConnectionNamesConfigurationSchema = joi_1.default.object({
    connection_names: exports.MapOfObjectsSchema.required(),
    default_connection_name: exports.ConnectionNamesSchema.required(),
});
