// @shipengine/connect-carrier-api/lib/app/metadata/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredProperty = exports.ServiceGrade = exports.ServiceClass = exports.SupportedLabelSize = exports.ShippingServiceAttribute = exports.CarrierAttribute = void 0;
var CarrierAttribute;
(function (CarrierAttribute) {
    CarrierAttribute["ManifestDigital"] = "ManifestDigital";
    CarrierAttribute["ManifestPhysical"] = "ManifestPhysical";
    CarrierAttribute["Consolidator"] = "Consolidator";
    CarrierAttribute["Regional"] = "Regional";
    CarrierAttribute["RequiresInboundWebhook"] = "RequiresInboundWebhook";
})(CarrierAttribute || (exports.CarrierAttribute = CarrierAttribute = {}));
var ShippingServiceAttribute;
(function (ShippingServiceAttribute) {
    ShippingServiceAttribute["Returns"] = "Returns";
    ShippingServiceAttribute["MultiPackage"] = "MultiPackage";
    ShippingServiceAttribute["Tracking"] = "Tracking";
    ShippingServiceAttribute["ConsolidatorService"] = "ConsolidatorService";
    ShippingServiceAttribute["AutomatedTrackingAllowed"] = "AutomatedTrackingAllowed";
    ShippingServiceAttribute["ManifestDigital"] = "ManifestDigital";
    ShippingServiceAttribute["ManifestPhysical"] = "ManifestPhysical";
    ShippingServiceAttribute["SameDayService"] = "SameDayService";
    ShippingServiceAttribute["Tip"] = "Tip";
    ShippingServiceAttribute["DeliveryWindow"] = "DeliveryWindow";
    ShippingServiceAttribute["PickupOnLabelCreation"] = "PickupOnLabelCreation";
    ShippingServiceAttribute["PaperlessLabel"] = "PaperlessLabel";
    ShippingServiceAttribute["InboundTrackingUpdates"] = "InboundTrackingUpdates";
})(ShippingServiceAttribute || (exports.ShippingServiceAttribute = ShippingServiceAttribute = {}));
var SupportedLabelSize;
(function (SupportedLabelSize) {
    SupportedLabelSize["Inches4x6"] = "Inches4x6";
    SupportedLabelSize["Inches4x8"] = "Inches4x8";
})(SupportedLabelSize || (exports.SupportedLabelSize = SupportedLabelSize = {}));
var ServiceClass;
(function (ServiceClass) {
    ServiceClass["Unspecified"] = "Unspecified";
    ServiceClass["Ground"] = "Ground";
    ServiceClass["OneDay"] = "OneDay";
    ServiceClass["OneDayEarly"] = "OneDayEarly";
    ServiceClass["OneDayEarlyAm"] = "OneDayEarlyAm";
    ServiceClass["TwoDay"] = "TwoDay";
    ServiceClass["TwoDayEarly"] = "TwoDayEarly";
    ServiceClass["ThreeDay"] = "ThreeDay";
})(ServiceClass || (exports.ServiceClass = ServiceClass = {}));
var ServiceGrade;
(function (ServiceGrade) {
    ServiceGrade["Unspecified"] = "Unspecified";
    ServiceGrade["Economy"] = "Economy";
    ServiceGrade["Expedited"] = "Expedited";
    ServiceGrade["Overnight"] = "Overnight";
    ServiceGrade["Standard"] = "Standard";
})(ServiceGrade || (exports.ServiceGrade = ServiceGrade = {}));
var RequiredProperty;
(function (RequiredProperty) {
    RequiredProperty["Weight"] = "Weight";
    RequiredProperty["Dimensions"] = "Dimensions";
})(RequiredProperty || (exports.RequiredProperty = RequiredProperty = {}));
