// @shipengine/connect-carrier-api/lib/app/metadata/package-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageTypeSchema = exports.PackageAttribute = exports.RequiredToShipEnum = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const connect_runtime_1 = require("@shipengine/connect-runtime");
var RequiredToShipEnum;
(function (RequiredToShipEnum) {
    RequiredToShipEnum["Weight"] = "Weight";
    RequiredToShipEnum["Dimensions"] = "Dimensions";
})(RequiredToShipEnum || (exports.RequiredToShipEnum = RequiredToShipEnum = {}));
var PackageAttribute;
(function (PackageAttribute) {
    PackageAttribute["International"] = "International";
    PackageAttribute["Domestic"] = "Domestic";
    PackageAttribute["Consolidator"] = "Consolidator";
})(PackageAttribute || (exports.PackageAttribute = PackageAttribute = {}));
exports.PackageTypeSchema = joi_1.default.object({
    Id: joi_1.default.string().uuid().required(),
    Name: joi_1.default.string().required().max(50),
    ApiCode: joi_1.default.string().optional().pattern(connect_runtime_1.ApiCodeRegex, connect_runtime_1.ApiCodeValidationMessage),
    CarrierPackageTypeCode: joi_1.default.string().required().max(50),
    Description: joi_1.default.string().optional().max(500),
    Abbreviation: joi_1.default.string().optional().max(20),
    PackageAttributes: joi_1.default.array()
        .required()
        .items(joi_1.default.string().valid(...Object.values(PackageAttribute))),
    RequiredToShip: joi_1.default.array()
        .optional()
        .items(joi_1.default.string().valid(...Object.values(RequiredToShipEnum))),
});
