// @shipengine/connect-carrier-api/lib/app/metadata/services-attributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAttributesEnumSchema = exports.ServiceAttributesEnum = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
var ServiceAttributesEnum;
(function (ServiceAttributesEnum) {
    ServiceAttributesEnum["Returns"] = "Returns";
    ServiceAttributesEnum["MultiPackage"] = "MultiPackage";
    ServiceAttributesEnum["Tracking"] = "Tracking";
    ServiceAttributesEnum["ConsolidatorService"] = "ConsolidatorService";
    ServiceAttributesEnum["AutomatedTrackingAllowed"] = "AutomatedTrackingAllowed";
    ServiceAttributesEnum["ManifestDigital"] = "ManifestDigital";
    ServiceAttributesEnum["ManifestPhysical"] = "ManifestPhysical";
    ServiceAttributesEnum["SameDayService"] = "SameDayService";
    ServiceAttributesEnum["Tip"] = "Tip";
    ServiceAttributesEnum["DeliveryWindow"] = "DeliveryWindow";
    ServiceAttributesEnum["PickupOnLabelCreation"] = "PickupOnLabelCreation";
    ServiceAttributesEnum["PaperlessLabel"] = "PaperlessLabel";
    ServiceAttributesEnum["InboundTrackingUpdates"] = "InboundTrackingUpdates";
})(ServiceAttributesEnum || (exports.ServiceAttributesEnum = ServiceAttributesEnum = {}));
exports.ServiceAttributesEnumSchema = joi_1.default.string().valid(...Object.values(ServiceAttributesEnum));
