// @shipengine/connect-carrier-api/lib/app/metadata/shipping-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShippingServiceSchema = void 0;
const tslib_1 = require("tslib");
const confirmation_type_1 = require("./confirmation-type");
const country_association_1 = require("./country-association");
const services_attributes_1 = require("./services-attributes");
const label_sizes_1 = require("./label-sizes");
const service_class_1 = require("./service-class");
const service_grade_1 = require("./service-grade");
const service_required_properties_1 = require("./service-required-properties");
const joi_1 = tslib_1.__importDefault(require("joi"));
const connect_runtime_1 = require("@shipengine/connect-runtime");
exports.ShippingServiceSchema = joi_1.default.object({
    ConfirmationTypes: joi_1.default.array().optional().items(confirmation_type_1.ConfirmationTypeSchema).unique('Type'),
    ServiceAttributes: joi_1.default.array().items(services_attributes_1.ServiceAttributesEnumSchema).optional(),
    SupportedCountries: joi_1.default.array().optional().items(country_association_1.CountryAssociationSchema).unique('FromCountry'),
    SupportedLabelSizes: joi_1.default.array().optional().items(label_sizes_1.LabelSizesEnumSchema).unique(),
    RequiredProperties: joi_1.default.array().optional().items(service_required_properties_1.ServiceRequiredPropertiesEnumSchema).unique(),
    Grade: service_grade_1.ServiceGradeEnumSchema.optional(),
    Class: service_class_1.ServiceClassEnumSchema.optional(),
    LabelCode: joi_1.default.string().optional().max(50),
    International: joi_1.default.boolean().strict().optional(),
    Code: joi_1.default.string().required().max(50),
    Abbreviation: joi_1.default.string().required().max(20),
    Name: joi_1.default.string().required().max(80),
    ApiCode: joi_1.default.string().optional().pattern(connect_runtime_1.ApiCodeRegex, connect_runtime_1.ApiCodeValidationMessage),
    Id: joi_1.default.string().uuid().required(),
});
