// @shipengine/connect-carrier-api/lib/app/metadata/shipping-option.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShippingOptionDictionarySchema = exports.ShippingOptionEnum = exports.ShippingOptionDescriptionSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
exports.ShippingOptionDescriptionSchema = joi_1.default.object({
    Name: joi_1.default.string().required().max(50),
    Description: joi_1.default.string().required().max(255),
}).optional();
var ShippingOptionEnum;
(function (ShippingOptionEnum) {
    ShippingOptionEnum["ContainsAlcohol"] = "alcohol";
    ShippingOptionEnum["B13ACanada"] = "b13a-canada";
    ShippingOptionEnum["BillToSender"] = "bill-to-sender";
    ShippingOptionEnum["BillToThirdParty"] = "bill-to-third-party";
    ShippingOptionEnum["CollectOnDelivery"] = "collect-on-delivery";
    ShippingOptionEnum["ConsequentialLoss"] = "consequential-loss";
    ShippingOptionEnum["DangerousGoods"] = "dangerous-goods";
    ShippingOptionEnum["DeliveryMessage"] = "delivery-message";
    ShippingOptionEnum["DontPrepayPostage"] = "dont-prepay-postage";
    ShippingOptionEnum["DryIce"] = "dry-ice";
    ShippingOptionEnum["EmailNotification"] = "email-notification";
    ShippingOptionEnum["FreightClass"] = "freight-class";
    ShippingOptionEnum["HoldForPickup"] = "hold-for-pickup";
    ShippingOptionEnum["IncludeReturnLabel"] = "include-return-label";
    ShippingOptionEnum["LocalCollect"] = "local-collect";
    ShippingOptionEnum["NonMachinable"] = "non-machinable";
    ShippingOptionEnum["NotificationType"] = "notification-type";
    ShippingOptionEnum["ReleaseNoSignature"] = "shipper-release";
    ShippingOptionEnum["RequiresAdditionalHandling"] = "additional-handling";
    ShippingOptionEnum["Safeplace"] = "safeplace";
    ShippingOptionEnum["SaturdayDelivery"] = "saturday-delivery";
    ShippingOptionEnum["SaturdayGuarantee"] = "saturday-guarantee";
    ShippingOptionEnum["SmsNotification"] = "sms-notification";
    ShippingOptionEnum["SpecialHandling"] = "special-handling";
    ShippingOptionEnum["ThirdPartyConsignee"] = "third-party-consignee";
    ShippingOptionEnum["CarrierInsurance"] = "carrier-insurance";
})(ShippingOptionEnum || (exports.ShippingOptionEnum = ShippingOptionEnum = {}));
exports.ShippingOptionDictionarySchema = joi_1.default.object(Object.values(ShippingOptionEnum).reduce((accumulator, option) => {
    accumulator[option] = exports.ShippingOptionDescriptionSchema;
    return accumulator;
}, {}));
