// @shipengine/connect-carrier-api/lib/models/customs/customs-schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsSchema = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const customs_content_types_schema_1 = require("./customs-content-types-schema");
const customs_non_delivery_schema_1 = require("./customs-non-delivery-schema");
const customs_item_schema_1 = require("./customs-item-schema");
const currency_schema_1 = require("../currency-schema");
const terms_of_trade_code_schema_1 = require("../inconterms/terms-of-trade-code-schema");
const importer_of_record_schema_1 = require("../importer-records/importer-of-record-schema");
const customs_charges_schema_1 = require("./customs-charges-schema");
exports.CustomsSchema = joi_1.default.object({
    contents: customs_content_types_schema_1.CustomsContentTypesSchema.optional(),
    non_delivery: customs_non_delivery_schema_1.CustomsNonDeliverySchema.optional(),
    customs_items: joi_1.default.array().items(customs_item_schema_1.CustomsItemSchema).required(),
    buyer_shipping_amount_paid: currency_schema_1.CurrencySchema.optional(),
    duties_paid: currency_schema_1.CurrencySchema.optional(),
    declaration: joi_1.default.string().optional(),
    termsOfTradeCode: terms_of_trade_code_schema_1.TermsOfTradeCodeSchema.optional(),
    importer_of_record: importer_of_record_schema_1.ImporterOfRecordSchema.optional(),
    invoice_additional_details: customs_charges_schema_1.CustomsChargesSchema.optional(),
    contents_explanation: joi_1.default.string().optional(),
});
