// @shipengine/connect-carrier-api/lib/models/customs/customs-content-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsContentTypes = void 0;
/** @description Denotes the type of content for customs  */
var CustomsContentTypes;
(function (CustomsContentTypes) {
    /** @description This is generalized merchandise that someone has purchased */
    CustomsContentTypes["Merchandise"] = "merchandise";
    /** @description Documents that are being shipped */
    CustomsContentTypes["Documents"] = "documents";
    /** @description An item that is being shipped as a gift (not purchased) */
    CustomsContentTypes["Gift"] = "gift";
    /** @description Sample goods that are being shipped */
    CustomsContentTypes["Sample"] = "sample";
    /** @description Items that are being returned */
    CustomsContentTypes["ReturnedGoods"] = "returned_goods";
    /** @description   What other means in case of the shipment*/
    CustomsContentTypes["Other"] = "other";
})(CustomsContentTypes || (exports.CustomsContentTypes = CustomsContentTypes = {}));
