// connect-carrier-api-utils/lib/third-party-connector/request/third-party-request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdPartyRequest = void 0;
/**
 * Unified class to prepare request object for third party API.
 */
class ThirdPartyRequest {
    constructor() {
        /**
         * Set the request Url.
         * @param url
         * @constructor
         */
        this.Url = (url) => {
            this.ThirdPartyRequestProvider.url = url;
            return this;
        };
        /**
         * Set the request header.
         * @param header
         * @constructor
         */
        this.Header = (header) => {
            if (!this.ThirdPartyRequestProvider.headers) {
                this.ThirdPartyRequestProvider.headers = {};
            }
            const tmpHeaders = this.ThirdPartyRequestProvider.headers;
            Object.keys(header).forEach((key) => {
                tmpHeaders[key] = header[key];
            });
            this.ThirdPartyRequestProvider.headers = tmpHeaders;
            return this;
        };
        /**
         * Set the request headers.
         * @param headers
         * @constructor
         */
        this.Headers = (headers) => {
            headers.forEach((header) => {
                this.Header(header);
            });
            return this;
        };
        /**
         * Set the request method.
         * @param method
         * @constructor
         */
        this.Method = (method) => {
            this.ThirdPartyRequestProvider.method = method;
            return this;
        };
        /**
         * Set the request body.
         * @param body
         * @constructor
         */
        this.Body = (body) => {
            this.ThirdPartyRequestProvider.data = body;
            return this;
        };
        /**
         * Set the request query.
         * @param queryParam
         * @constructor
         */
        this.Query = (queryParam) => {
            this.ThirdPartyRequestProvider.url += `?${queryParam.key}=${queryParam.value}`;
            return this;
        };
        /**
         * Set the request queries.
         * @param queryParam
         * @constructor
         */
        this.Queries = (queryParam) => {
            let queryString = '?';
            queryParam.forEach((query) => {
                queryString += `${query.key}=${query.value}&`;
            });
            this.ThirdPartyRequestProvider.url += queryString;
            return this;
        };
        this.ThirdPartyRequestProvider = {};
    }
}
exports.ThirdPartyRequest = ThirdPartyRequest;
